% Preamble
\documentclass{article}

%Dependencies
\usepackage[left]{lineno}
%\usepackage{indentfirst}
\usepackage{titlesec}
\usepackage[utf8]{inputenc}
\usepackage{amsmath}
\usepackage{amsfonts}
\usepackage{amssymb}
\usepackage{color,soul}
%\usepackage{times}
\usepackage[sc]{mathpazo} %Like Palatino with extensive math support
\linespread{1.25}
% Default margins are too wide all the way around. I reset them here
\usepackage{fullpage}
% Force figures to be in correct section
\usepackage{placeins}


% Fonts and language
\RequirePackage[utf8]{inputenc}
\RequirePackage[english]{babel}
\RequirePackage{amsmath,amsfonts,amssymb}
\RequirePackage{mathpazo}
\RequirePackage[scaled]{helvet}
\RequirePackage[T1]{fontenc}
\RequirePackage{url}
\RequirePackage[colorlinks=true, allcolors=black]{hyperref}
%\RequirePackage[colorlinks=true, allcolors=blue,draft=true]{hyperref}
\RequirePackage{lettrine}
\RequirePackage{cleveref}

% Graphics, tables and other formatting
\RequirePackage{graphicx,xcolor}
\RequirePackage{colortbl}
\RequirePackage{booktabs}
\RequirePackage{tikz}
\RequirePackage{algorithm}
\RequirePackage[noend]{algpseudocode}
\RequirePackage{changepage}
\RequirePackage[labelfont={bf,sf},%
                labelsep=space,%
                figurename=Figure,%
                singlelinecheck=off,%
                justification=RaggedRight]{caption}
%\setlength{\columnsep}{24pt} % Distance between the two columns of text
\setlength{\parindent}{12pt} % Paragraph indent

% Bibliography
\usepackage{natbib} \bibpunct{(}{)}{;}{author-year}{}{,}
\bibliographystyle{amnatnat}
\addto{\captionsenglish}{\renewcommand{\refname}{Literature Cited}}
\setlength{\bibsep}{0.0pt}

% Running headers
\usepackage{fancyhdr}
\setlength{\headheight}{28pt}

% Graphics package
\usepackage{graphicx}
\graphicspath{{../../output/figures/}.pdf}

% New commands: fonts
%\newcommand{\code}{\fontfamily{pcr}\selectfont}
%\newcommand*\chem[1]{\ensuremath{\mathrm{#1}}}
\newcommand\numberthis{\addtocounter{equation}{1}\tag{\theequation}}

%% Other Options

% Change subsection numbering
\renewcommand\thesubsection{\arabic{subsection})}
\renewcommand\thesubsubsection{}

% Subsubsection Title Formatting
\titleformat{\subsubsection}    
{\normalfont\fontsize{12pt}{17}\itshape}{\thesubsubsection}{12pt}{}


%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\title{Supplementary Material (Appendices A--E) for: Sexually antagonistic variation and the evolution of dimorphic sexual systems. \textit{The American Naturalist}}

\author{Colin Olito$^{\ast,1,2}$ \& Tim Connallon$^{1}$}
\date{\today}

\begin{document}
\maketitle


\noindent{} $^{1}$ Centre for Geometric Biology, School of Biological Sciences, Monash University, Victoria 3800, Australia.

\noindent{} $^{2}$ \textit{Current address}: Department of Biology, Section for Evolutionary Ecology, Lund University, Lund 223 62, Sweden.

\noindent{} $^{\ast}$ Corresponding author e-mail: \url{colin.olito@gmail.com}

\newpage
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
% Running Header
\pagestyle{fancyplain}
\makeatother
\lhead{\textit{Supplement to Olito \& Connallon. SA variation and dimorphic sexual systems. \textit{Am.~Nat.}.\\}}
\lhead{\textit{Supplement to Olito \& Connallon (2019). \textit{Am.~Nat.}\\ }}
\rhead{\textit{Evolution of dimorphic sexual systems}\\ }
\renewcommand{\headrulewidth}{0pt}
\renewcommand{\footrulewidth}{0pt}
\addtolength{\headheight}{0pt}


%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\section*{Appendix A: Development of the recursions}
\renewcommand{\theequation}{A\arabic{equation}}
\titleformat{\subsubsection}    
{\normalfont\fontsize{12pt}{17}\itshape}{\thesubsubsection}{12pt}{}

%\setcounter{subsection}{0}  % reset counter 
%\setcounter{equation}{0}  % reset counter 

Here, we fully develop the system of recursion equations underlying each of the four models described in the main text: the evolution of gynodioecy via invasion of a (1) completely dominant, or (2) completely recessive male-sterility allele; and the evolution of androdioecy via the invasion of a (3) completely dominant or (4) completely recessive female-sterility allele. We start by briefly re-iterating the main assumptions of the models. We then describe the calculation of adult frequencies after inbreeding depression, as this is the same for all four models. We then walk through the derivation of each of the four models in turn. Where possible, we highlight only essential differences between models rather than re-deriving identical equations. \bigskip

\subsection*{Assumptions}

Consider the genetic system described in the main text -- two diallelic autosomal loci, $\mathbf{A}$ (with alleles $A$, $a$) and $\mathbf{M}$ (with alleles $M_1$, $M_2$), that recombine at rate $r$ per meiosis, in a large, initially hermaphroditic, population. Assume that the $\mathbf{A}$ locus is under sexually antagonistic selection, with the $A$ allele having female-beneficial (male-deleterious) fitness effects, and the $a$ allele having male-beneficial (female-deleterious) fitness effects. Assume also that the $M_1$ allele at the $\mathbf{M}$ locus has a relative fitness of $1$ for both sex-functions, while the $M_2$ allele causes complete sterility through one sex function (e.g., via cessation of production, or the production of inviable pollen or ovules). The population rate of self-fertilization, $C$, is assumed to be independent of the genotype at the $\mathbf{A}$ locus, while the genotype at the $\mathbf{M}$ locus determines whether or not a genotype is capable of self-fertilization. We also assume a constant population level of inbreeding depression, $\delta$, and that a vanishingly small amount of pollen is adequate for successful self-fertilization. Generations are assumed to be non-overlapping, and the life-cycle for all models proceeds as follows: birth $\rightarrow$ viability seleciton on self-fertilized zygotes due to inbreeding depression $\rightarrow$ selection on gamete production $\rightarrow$ fertilization.

\subsection*{Adult genotypic frequencies after inbreeding depression}

There are four possible combinations of alleles at the $\mathbf{A}$ and $\mathbf{M}$ loci that can be found on any chromosome: $A M_1$, $A M_2$, $a M_1$, and $a M_2$. Let $x_i$ and $y_i$ denote the haplotype frequencies $[AM_1,AM_2,aM_1,aM_2]$ in female and male gametes respectively. The calculation of adult genotypic frequencies after inbreeding is identical for all models. Let $F_{ij}$ denote the frequency among offspring of the genotype formed by the union of the \textit{i}th and \textit{j}th gametic haplotypes via outcrossing. Similarly, let $G_{ij}$ denote the frequency among offspring of the genotype formed by the union of the \textit{i}th and \textit{j}th gametic haplotypes via self-fertilization. We assume no parent-of-origin effects on fitness, and so index the ten possible pairs of gametic haplotypes where $i \geq j$ hereafter. The frequency of each genotype among adults after accounting for inbreeding depression, $F^A_{ij}$, will then be:
\begin{equation} \label{eq:AdultFreq}
	F^A_{ij} = \frac{F_{ij} + G_{ij}(1 - \delta)}{\overline{D}},
\end{equation}

\noindent where 
\begin{equation} \label{eq:Dbar}
\overline{D} = \sum_{i=1}^{4}\sum_{j=i}^{4} F_{ij} + \sum_{i=1}^{4}\sum_{j=i}^{4} G_{ij}(1 - \delta).
\end{equation}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\subsection*{Gynodioecy via dominant male-sterility mutation}

For the model of the evolution of gynodioecy via the invasion of a completely dominant male-sterility mutation at $\mathbf{M}$, the successful invasion of the $M_2$ allele, either alone or coupled with an allele at the $\mathbf{A}$ locus as a haplotype, represents the evolution of gynodioecy from simultaneous hermaphroditism. The fitness expressions for adult genotype $ij$ through the female and male sex-function can be expressed in $4 \times 4$ matrices, with elements $w^f_{ij}$ (for female-function), and $w^m_{ij}$ (for male-function), yielding Table 1 from the main text:

\begin{table}[ht!]
\caption{Fitness expressions for diploid adults prior to reproduction for the model of a dominant male-sterility mutation ($w^f_{ij}$ denotes fitness effects through the female sex-function , $w^m_{ij}$ for male sex-function).}
\centering
\begin{tabular}{l c c c c} \hline
Haplotype & $ AM_1$ & $ AM_2$ & $ aM_1$ & $ aM_2$ \\
\hline
Female-function & & & & \\
$ AM_1$ & $1$ & $(1 + k)$ & $(1 - h_f s_f)$        & $(1 - h_f s_f)(1 + k)$ \\
$ AM_2$ & $-$ & $(1 + k)$ & $(1 - h_f s_f)(1 + k)$ & $(1 - h_f s_f)(1 + k)$ \\
$ aM_1$ & $-$ & $-$       & $(1 - s_f)$            & $(1 - s_f)(1 + k)$ \\
$ aM_2$ & $-$ & $-$       & $-$                    & $(1 - s_f)(1 + k)$ \\
Male-function & & & & \\
$ AM_1$ & $(1 - s_m)$ & $0$ & $(1 - h_m s_m)$ & $0$ \\
$ AM_2$ & $-$         & $0$ & $0$             & $0$ \\
$ aM_1$ & $-$         & $-$ & $1$             & $0$ \\
$ aM_2$ & $-$         & $-$ & $-$             & $0$ \\
\hline
\end{tabular}
\bigskip{} \\
{\footnotesize Note: Rows and columns indicate the \textit{i}th and \textit{j}th gametic haplotype respectively. The lower triangle of each matrix is the reflection of the upper triangle, and is omitted for simplicity and consistency with the $i \geq j$ row/column indexing used throughout the article.}
\end{table}

\noindent The total number of ovules produced by the population at reproduction will be proportional to
\begin{equation} 
O_T = \sum_{i=1}^{4}\sum_{j=i}^{4} F^A_{ij} w^f_{ij}.
\end{equation}

\noindent Because $M_2$ is assumed to be completely dominant, only those genotypes that are homozygous for $M_1$ are phenotypically hermaphrodite, and therefore able to self-fertilize. The proportion of ovules that are self-fertilized will therefore be proportional to
\begin{equation} 
S = C \bigg( \frac{F^A_{11} w^f_{11} + F^A_{13} w^f_{13} + F^A_{33} w^f_{33}}{O_T}\bigg).
\end{equation}

\noindent The proportion of ovules fertilized via outcrossing is the complement of $S$, and will be proportional to
\begin{equation} 
(1 - S) = \bigg( \frac{(1 - C)(F^A_{11} w^f_{11} + F^A_{13} w^f_{13} + F^A_{33} w^f_{33}) + U}{O_T}\bigg),
\end{equation}

\noindent where $U = F^A_{12} w^f_{12} + F^A_{14} w^f_{14} + F^A_{22} w^f_{22} + F^A_{23} w^f_{23} + F^A_{24} w^f_{24} + F^A_{34} w^f_{34} + F^A_{44} w^f_{44}$ is the total frequency of unisexual females (all genotypes either heterozygous or homozygous for $M_2$). 

Only the three genotypes that are homozygous for $M_1$ are phenotypically hermaphrodite and able to self-fertilize. The proportional contribution of self-fertilized offspring by each of these genotypes are then:
\begin{align*} \label{eq:DomSelfOvules}
    o^S_{11} &= C \bigg(\frac{F^A_{11} w^f_{11}}{O^S_T} \bigg) \\
    o^S_{13} &= C \bigg(\frac{F^A_{13} w^f_{13}}{O^S_T} \bigg) \\
    o^S_{33} &= C \bigg(\frac{F^A_{33} w^f_{33}}{O^S_T} \bigg), \numberthis
\end{align*}

\noindent where $O^S_T = C(F^A_{11} w^f_{11} + F^A_{13} w^f_{13} + F^A_{33} w^f_{33})$, and simplify down to the relative genotypic frequencies after selection. $o^S_{ij} = 0$ for all genotypes that are phenotypically unisexual (female). The relevant genotypic recursions (Eq(2) in the main text) describing the frequency in the next generation of each genotype among zygotes derived from self-fertilization are then:
\begin{align*} \label{eq:DomSelfRecs}
    G'_{11} &= S (o^S_{11} + o^S_{13}/4) \\
    G'_{13} &= S (o^S_{13}/2) \\
    G'_{33} &= S (o^S_{33} + o^S_{13}/4), \numberthis
\end{align*}

\noindent and all $G'_{ij} = 0$ where $ij \neq [11,13,33]$.

The genotypic frequency recursions for offpsring derived via outcrossing follow similar logic. The proporitonal contribution of each genotype to the pool of outcrossed ovules are
\begin{align*}
    o^X_{11} &= (1 - C) (F^A_{11} w^f_{11})/O^X_T \\
    o^X_{12} &= (F^A_{12} w^f_{12})/O^X_T \\
    o^X_{13} &= (1 - C) (F^A_{13} w^f_{13})/O^X_T \\
    o^X_{14} &= (F^A_{14} w^f_{14})/O^X_T \\
    o^X_{22} &= (F^A_{22} w^f_{22})/O^X_T \\
    o^X_{23} &= (F^A_{23} w^f_{23})/O^X_T \\
    o^X_{24} &= (F^A_{24} w^f_{24})/O^X_T \\
    o^X_{33} &= (1 - C) (F^A_{33} w^f_{33})/O^X_T \\
    o^X_{34} &= (F^A_{34} w^f_{34})/O^X_T \\
    o^X_{44} &= (F^A_{44} w^f_{44})/O^X_T, \numberthis
\end{align*}

\noindent where $O^X_T = (1 - S)O_T$. Only those genotypes that are phenotypically hermaphrodite are able to sire outcross offspring. The proportional contribution of each genotype  that is phenotypically hermaphrodite to the pool of outcrossed pollen is equal to the relative genotypic frequencies after selection:
\begin{align*}
    p^X_{11} &= (F^A_{11} w^m_{11}) / P^X_T \\
    p^X_{13} &= (F^A_{13} w^m_{13}) / P^X_T \\
    p^X_{33} &= (F^A_{33} w^m_{33}) / P^X_T, \numberthis
\end{align*}

\noindent where $P^X_T = F^A_{11} w^m_{11} + F^A_{13} w^m_{13} + F^A_{33} w^m_{33}$. From the relative genotypic contributions to the pool of outcrossed ovules ($o^X_{ij}$) and pollen ($p^X_{ij}$), we can then calculate the haplotype frequencies among outcrossed ovules ($x_i$) and pollen ($y_i$) following standard two-locus theory. The haplotype frequencies among outcrossed ovules, $x_i$, are
\begin{align*} 
    x_{1} &= o^X_{11} + \frac{o^X_{12} + o^X_{13} + o^X_{14}}{2} - r \bigg( \frac{o^X_{14} - o^X_{23}}{2} \bigg)  \\
    x_{2} &= o^X_{22} + \frac{o^X_{12} + o^X_{23} + o^X_{24}}{2} + r \bigg( \frac{o^X_{14} - o^X_{23}}{2} \bigg)  \\
    x_{3} &= o^X_{33} + \frac{o^X_{13} + o^X_{23} + o^X_{34}}{2} + r \bigg( \frac{o^X_{14} - o^X_{23}}{2} \bigg)  \\
    x_{4} &= o^X_{44} + \frac{o^X_{14} + o^X_{24} + o^X_{34}}{2} - r \bigg( \frac{o^X_{14} - o^X_{23}}{2} \bigg),  \numberthis
\end{align*}

\noindent and the haplotype frequencies among the pool of outcrossed pollen, $y_i$, are
\begin{align*} 
    y_{1} &= p^X_{11} + p^X_{13}/2  \\
    y_{2} &= 0  \\
    y_{3} &= p^X_{33} + p^X_{13}/2  \\
    y_{4} &= 0.  \numberthis
\end{align*}

\noindent The genotypic recursions describing the frequency in the next generation of each genotype among zygotes derived from outcrossing are calculated from these haplotype frequences as:
\begin{align*} 
     F'_{11} &= (1 - S) (x_1 y_1)  \\
    F'_{12} &= (1 - S) (x_2 y_1)  \\
    F'_{13} &= (1 - S) (x_1 y_3 + x_3 y_1)  \\
    F'_{14} &= (1 - S) (x_4 y_1)  \\
    F'_{22} &= 0 \\
    F'_{23} &= (1 - S) (x_2 y_3)  \\
    F'_{24} &= 0 \\
    F'_{33} &= (1 - S) (x_3 y_3)  \\
    F'_{34} &= (1 - S) (x_4 y_3)  \\
    F'_{44} &= 0. \numberthis
\end{align*}








%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\newpage{}
\subsection*{Gynodioecy via recessive male-sterility mutation}

For the model of the evolution of gynodioecy via the invasion of a completely recessive male-sterility mutation at $\mathbf{M}$, only genotypes that are homozygous for the $M_2$ allele are phenotypically female. As described above, the adult genotypic frequencies, $F^A_{ij}$, are described by Equations \ref{eq:AdultFreq} and \ref{eq:Dbar}. Now, the fitness expressions associated with each $F^A_{ij}$ through the female and male sex-function are:

\begin{table}[ht!]
\caption{Fitness expressions for diploid adults prior to reproduction for the model of a recessive male-sterility mutation ($w^f_{ij}$ denotes fitness effects through the female sex-function, $w^m_{ij}$ for male sex-function).}
\centering
\begin{tabular}{l c c c c} \hline
Haplotype & $ AM_1$ & $ AM_2$ & $ aM_1$ & $ aM_2$ \\
\hline
Female-function & & & & \\
$ AM_1$ & $1$ & $1$       & $(1 - h_f s_f)$ & $(1 - h_f s_f)$        \\
$ AM_2$ & $-$ & $(1 + k)$ & $(1 - h_f s_f)$ & $(1 - h_f s_f)(1 + k)$ \\
$ aM_1$ & $-$ & $-$       & $(1 - s_f)$     & $(1 - s_f)$            \\
$ aM_2$ & $-$ & $-$       & $-$             & $(1 - s_f)(1 + k)$     \\
Male-function & & & & \\
$ AM_1$ & $(1 - s_m)$ & $(1 - s_m)$ & $(1 - h_m s_m)$ & $(1 - h_m s_m)$ \\
$ AM_2$ & $-$         & $0$         & $(1 - h_m s_m)$ & $0$             \\
$ aM_1$ & $-$         & $-$         & $1$             & $1$             \\
$ aM_2$ & $-$         & $-$         & $-$             & $0$             \\
\hline
\end{tabular}
\bigskip{} \\
{\footnotesize Note: Rows and columns indicate the \textit{i}th and \textit{j}th gametic haplotype respectively. The lower triangle of each matrix is the reflection of the upper triangle, and is omitted for simplicity and consistency with the $i \geq j$ row/column indexing used throughout the article.}
\end{table}


\noindent The total number of ovules produced by the population at reproduction will again be proportional to
\begin{equation} 
O_T = \sum_{i=1}^{4}\sum_{j=i}^{4} F^A_{ij} w^f_{ij}.
\end{equation}

\noindent Since the $M_2$ allele is now assumed to be completely recessive, all genotypes not homozygous for $M_2$ are phenotypically hermaphrodite, and therefore able to self-fertilize. The proportion of ovules that are self-fertilized will therefore be proportional to
\begin{equation} 
S = C \bigg( \frac{F^A_{11} w^f_{11} + F^A_{12} w^f_{12} + F^A_{13} w^f_{13} + F^A_{14} w^f_{14} +
                   F^A_{23} w^f_{23} + F^A_{33} w^f_{33} + F^A_{34} w^f_{34}}{O_T}\bigg),
\end{equation}

\noindent and the proportion of ovules fertilized via outcrossing is the complement of $S$:
\begin{equation} 
(1 - S) = \bigg( \frac{(1 - C)(F^A_{11} w^f_{11} + F^A_{12} w^f_{12} + F^A_{13} w^f_{13} + F^A_{14} w^f_{14} +
                   F^A_{23} w^f_{23} + F^A_{33} w^f_{33} + F^A_{34} w^f_{34}) + U}{O_T}\bigg),
\end{equation}

\noindent where $U = F^A_{22} w^f_{22} + F^A_{24} w^f_{24} + F^A_{44} w^f_{44}$ is the total frequency of all unisexual females (all genotypes homozygous for $M_2$). 

The proportional contribution of each genotype to the pool of self-fertilized offspring is then:
\begin{align*} \label{eq:RecSelfOvules}
    o^S_{11} &= C \bigg(\frac{F^A_{11} w^f_{11}}{O^S_T} \bigg) \\
    o^S_{12} &= C \bigg(\frac{F^A_{12} w^f_{12}}{O^S_T} \bigg) \\
    o^S_{13} &= C \bigg(\frac{F^A_{13} w^f_{13}}{O^S_T} \bigg) \\
    o^S_{14} &= C \bigg(\frac{F^A_{14} w^f_{14}}{O^S_T} \bigg) \\
    o^S_{23} &= C \bigg(\frac{F^A_{23} w^f_{23}}{O^S_T} \bigg) \\
    o^S_{33} &= C \bigg(\frac{F^A_{33} w^f_{33}}{O^S_T} \bigg) \\
    o^S_{34} &= C \bigg(\frac{F^A_{34} w^f_{34}}{O^S_T} \bigg), \numberthis
\end{align*}

\noindent where $o^S_{ij} = 0$ where $ij \in [22,24,44]$, and $O^S_T = S \times O_T$. The resulting genotypic recursions describing the frequency in the next generation of each genotype among zygotes derived from self-fertilization are then:
\begin{align*} \label{eq:RecSelfRecs}
    G'_{11} &= S \Big(o^S_{11} + \frac{o^S_{12} + o^S_{13} + o^S_{14}(1-r)^2 + o^S_{23}r^2}{4} \Big) \\
    G'_{12} &= S \Big( \frac{o^S_{12} + o^S_{14}r(1-r) + o^S_{23}r(1-r)}{2} \Big) \\
    G'_{13} &= S \Big( \frac{o^S_{13} + o^S_{14}r(1-r) + o^S_{23}r(1-r)}{2} \Big) \\
    G'_{14} &= S \Big( \frac{o^S_{14}(1-r)^2 + o^S_{23}r^2}{2} \Big) \\
    G'_{22} &= S \Big( \frac{o^S_{12} + o^S_{13} + o^S_{14}r^2 + o^S_{23}(1-r)^2}{4} \Big) \\
    G'_{23} &= S \Big( \frac{o^S_{14}r^2 + o^S_{23}(1-r)^2}{2} \Big) \\
    G'_{24} &= S \Big( \frac{o^S_{14}r(1-r) + o^S_{23}r(1-r)}{2} \Big) \\
    G'_{33} &= S \Big(o^S_{33} + \frac{o^S_{13} + o^S_{14}r^2 + o^S_{23}(1-r)^2 + o^S_{34}}{4} \Big) \\
    G'_{34} &= S \Big( \frac{o^S_{14}r(1-r) + o^S_{23}r(1-r) + o^S_{34}}{2} \Big) \\
    G'_{44} &= S \Big( \frac{o^S_{34} + o^S_{14}(1-r)^2 + o^S_{23}r^2}{4} \Big). \numberthis
\end{align*}

The genotypic frequency recursions for offpsring derived via outcrossing are derived in similar fashion. The proporitonal contribution of each genotype to the pool of outcrossed ovules are
\begin{align*}
    o^X_{11} &= (1 - C) (F^A_{11} w^f_{11})/O^X_T \\
    o^X_{12} &= (1 - C) (F^A_{12} w^f_{12})/O^X_T \\
    o^X_{13} &= (1 - C) (F^A_{13} w^f_{13})/O^X_T \\
    o^X_{14} &= (1 - C) (F^A_{14} w^f_{14})/O^X_T \\
    o^X_{22} &= (F^A_{22} w^f_{22})/O^X_T \\
    o^X_{23} &= (1 - C) (F^A_{23} w^f_{23})/O^X_T \\
    o^X_{24} &= (F^A_{24} w^f_{24})/O^X_T \\
    o^X_{33} &= (1 - C) (F^A_{33} w^f_{33})/O^X_T \\
    o^X_{34} &= (1 - C) (F^A_{34} w^f_{34})/O^X_T \\
    o^X_{44} &= (F^A_{44} w^f_{44})/O^X_T, \numberthis
\end{align*}

\noindent where $O^X_T = (1 - S)O_T$. The proportional contribution of each genotype that is phenotypically hermaphrodite to the pool of outcrossed pollen is equal to the relative genotypic frequencies after selection:
\begin{align*}
    p^X_{11} &= (F^A_{11} w^f_{11}) / P^X_T \\
    p^X_{12} &= (F^A_{12} w^f_{12}) / P^X_T \\
    p^X_{13} &= (F^A_{13} w^f_{13}) / P^X_T \\
    p^X_{14} &= (F^A_{14} w^f_{14}) / P^X_T \\
    p^X_{23} &= (F^A_{23} w^f_{23}) / P^X_T \\
    p^X_{34} &= (F^A_{34} w^f_{34}) / P^X_T, \numberthis
\end{align*}

\noindent where $p^X_{ij} = 0$ where $ij \in [22,24,44]$, and $P^X_T = \sum_{i=1}^{4}\sum_{j=i}^{4} p^X_{ij}$. The haplotype frequencies among outcrossed ovules ($x_i$) are then:
\begin{align*} 
    x_{1} &= o^X_{11} + \frac{o^X_{12} + o^X_{13} + o^X_{14}}{2} - r \bigg( \frac{o^X_{14} - o^X_{23}}{2} \bigg)  \\
    x_{2} &= o^X_{22} + \frac{o^X_{12} + o^X_{23} + o^X_{24}}{2} + r \bigg( \frac{o^X_{14} - o^X_{23}}{2} \bigg)  \\
    x_{3} &= o^X_{33} + \frac{o^X_{13} + o^X_{23} + o^X_{34}}{2} + r \bigg( \frac{o^X_{14} - o^X_{23}}{2} \bigg)  \\
    x_{4} &= o^X_{44} + \frac{o^X_{14} + o^X_{24} + o^X_{34}}{2} - r \bigg( \frac{o^X_{14} - o^X_{23}}{2} \bigg),  \numberthis
\end{align*}

\noindent and the haplotype frequencies among the pool of outcrossed pollen ($y_i$) are:
\begin{align*} 
    y_{1} &= p^X_{11} + \frac{p^X_{12} + p^X_{13} + p^X_{14}}{2} - r \bigg( \frac{p^X_{14} - p^X_{23}}{2} \bigg)  \\
    y_{2} &= \frac{p^X_{12} p^X_{23}}{2} + r \bigg( \frac{p^X_{14} - p^X_{23}}{2} \bigg) \\
    y_{3} &= p^X_{33} + \frac{p^X_{13} + p^X_{23} + p^X_{34}}{2} + r \bigg( \frac{p^X_{14} - p^X_{23}}{2} \bigg)  \\
    y_{4} &= \frac{p^X_{14} p^X_{34}}{2} - r \bigg( \frac{p^X_{14} - p^X_{23}}{2} \bigg).  \numberthis
\end{align*}

\noindent The genotypic recursions describing the frequency in the next generation of each genotype among zygotes derived from outcrossing are calculated from these haplotype frequences as:
\begin{align*} \label{eq:GynRecOutRecs}
    F'_{11} &= (1 - S) (x_1 y_1)  \\
    F'_{12} &= (1 - S) (x_1 y_2 + x_2 y_1)  \\
    F'_{13} &= (1 - S) (x_1 y_3 + x_3 y_1)  \\
    F'_{14} &= (1 - S) (x_1 y_4 + x_4 y_1)  \\
    F'_{22} &= (1 - S) (x_2 y_2) \\
    F'_{23} &= (1 - S) (x_2 y_3 + x_3 y_2)  \\
    F'_{24} &= (1 - S) (x_2 y_4 + x_4 y_2) \\
    F'_{33} &= (1 - S) (x_3 y_3)  \\
    F'_{34} &= (1 - S) (x_3 y_4 + x_4 y_3)  \\
    F'_{44} &= (1 - S) (x_4 y_4). \numberthis
\end{align*}







%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\newpage{}
\subsection*{Androdioecy via dominant female-sterility mutation}

The model of Androdioecy via the invasion of a dominant female-sterility mutation is very similar in structure to the model of gynodioecy via a dominant male-sterility mutation. Adult genotypic frequencies again follow equations \ref{eq:AdultFreq} and \ref{eq:Dbar}. With a dominant female-sterility mutation, any genotype carrying the $M_2$ allele will be phenotypically male. The fitness expressions corresponding to each $F^A_{ij}$ are:


\begin{table}[ht!]
\caption{Fitness expressions for diploid adults prior to reproduction for the model of a dominant female-sterility mutation ($w^f_{ij}$ denotes fitness effects through the female sex-function , $w^m_{ij}$ for male sex-function).}
\centering
\begin{tabular}{l c c c c} \hline
Haplotype & $ AM_1$ & $ AM_2$ & $ aM_1$ & $ aM_2$ \\
\hline
Female-function & & & & \\
$ AM_1$ & $1$ & $0$ & $(1 - h_f s_f)$ & $0$ \\
$ AM_2$ & $-$ & $0$ & $0$             & $0$ \\
$ aM_1$ & $-$ & $-$ & $(1 - s_f)$     & $0$ \\
$ aM_2$ & $-$ & $-$ & $-$             & $0$ \\
Male-function & & & & \\
$ AM_1$ & $(1 - s_m)$ & $(1 - s_m)(1 + k)$ & $(1 - h_m s_m)$        & $(1 - h_m s_m)(1 + k)$ \\
$ AM_2$ & $-$         & $(1 - s_m)(1 + k)$ & $(1 - h_m s_m)(1 + k)$ & $(1 - h_m s_m)(1 + k)$ \\
$ aM_1$ & $-$         & $-$                & $1$                    & $(1 + k)$              \\
$ aM_2$ & $-$         & $-$                & $-$                    & $(1 + k)$              \\
\hline
\end{tabular}
\bigskip{} \\
{\footnotesize Note: Rows and columns indicate the \textit{i}th and \textit{j}th gametic haplotype respectively. The lower triangle of each matrix is the reflection of the upper triangle, and is omitted for simplicity and consistency with the $i \geq j$ row/column indexing used throughout the article.}
\end{table}

\noindent Because only phenotypically hermaphroditic individuals can produce ovules, the total number of ovules produced by the population at reproduction will be proportional to
\begin{equation} 
O_T = F^A_{11} w^f_{11} + F^A_{13} w^f_{13} + F^A_{33} w^f_{33}.
\end{equation}

\noindent The proportion of ovules that are self-fertilized will therefore be proportional to
\begin{equation} 
S = \frac{C (O_T)}{O_T},
\end{equation}

\noindent and the proportion of ovules fertilized via outcrossing is the complement of $S$:
\begin{equation} 
(1 - S) = \frac{(1 - C)O_T}{O_T}.
\end{equation}

Because only those genotypes homozygous for $M_1$ can self-fertilize, the structure of the recursions for selfed zygotes are the same as for the model of gynodioecy via invasion of a dominant male-sterility mutation. That is, the proportional contribution of each adult genotype to the pool of selfed ovules, $o^S_{ij}$ follow Eq(\ref{eq:DomSelfOvules}), and the resulting recursions, $G'_{ij}$ follow Eq(\ref{eq:DomSelfRecs}). However, the case is altered for outcrossed offspring. The proportional contributions to the pool of outcrossed ovules for each genotype are:

\begin{align*}
    o^X_{11} &= (1 - C) (F^A_{11} w^f_{11})/O^X_T \\
    o^X_{13} &= (1 - C) (F^A_{13} w^f_{13})/O^X_T \\
    o^X_{33} &= (1 - C) (F^A_{33} w^f_{33})/O^X_T, \numberthis
\end{align*}

\noindent where $O^X_T = (1 - S)O_T$. The proportional contribution of each genotype to the pool of outcrossed pollen after selection is:
\begin{align*}
    p^X_{11} &= (F^A_{11} w^f_{11}) / P^X_T \\
    p^X_{12} &= (F^A_{12} w^f_{12}) / P^X_T \\
    p^X_{13} &= (F^A_{13} w^f_{13}) / P^X_T \\
    p^X_{14} &= (F^A_{14} w^f_{14}) / P^X_T \\
    p^X_{22} &= (F^A_{22} w^f_{22}) / P^X_T \\
    p^X_{23} &= (F^A_{23} w^f_{23}) / P^X_T \\
    p^X_{24} &= (F^A_{24} w^f_{24}) / P^X_T \\
    p^X_{33} &= (F^A_{33} w^f_{33}) / P^X_T \\
    p^X_{34} &= (F^A_{34} w^f_{34}) / P^X_T \\
    p^X_{44} &= (F^A_{44} w^f_{44}) / P^X_T, \numberthis
\end{align*}

\noindent where $P^X_T = \sum_{i=1}^{4}\sum_{j=i}^{4} p^X_{ij}$. The resulting haplotype frequencies among outcrossed ovules ($x_i$) and pollen ($y_i$) are then:
\begin{align*} 
    x_{1} &= o^X_{11} + o^X_{13}/2  \\
    x_{2} &= 0  \\
    x_{3} &= o^X_{33} + o^X_{13}/2  \\
    x_{4} &= 0.  \numberthis
\end{align*}

\noindent and the haplotype frequencies among the pool of outcrossed pollen are
\begin{align*} 
    x_{1} &= p^X_{11} + \frac{p^X_{12} + p^X_{13} + p^X_{14}}{2} - r \bigg( \frac{p^X_{14} - p^X_{23}}{2} \bigg)  \\
    x_{2} &= p^X_{22} + \frac{p^X_{12} + p^X_{23} + p^X_{24}}{2} + r \bigg( \frac{p^X_{14} - p^X_{23}}{2} \bigg)  \\
    x_{3} &= p^X_{33} + \frac{p^X_{13} + p^X_{23} + p^X_{34}}{2} + r \bigg( \frac{p^X_{14} - p^X_{23}}{2} \bigg)  \\
    x_{4} &= p^X_{44} + \frac{p^X_{14} + p^X_{24} + p^X_{34}}{2} - r \bigg( \frac{p^X_{14} - p^X_{23}}{2} \bigg),  \numberthis
\end{align*}

\noindent The genotypic recursions describing the frequency in the next generation of each genotype among zygotes derived from outcrossing are calculated from these haplotype frequences as:
\begin{align*} 
    F'_{11} &= (1 - S) (x_1 y_1)  \\
    F'_{12} &= (1 - S) (x_1 y_2)  \\
    F'_{13} &= (1 - S) (x_1 y_3 + x_3 y_1)  \\
    F'_{14} &= (1 - S) (x_1 y_4)  \\
    F'_{22} &= 0 \\
    F'_{23} &= (1 - S) (x_3 y_2)  \\
    F'_{24} &= 0 \\
    F'_{33} &= (1 - S) (x_3 y_3)  \\
    F'_{34} &= (1 - S) (x_3 y_4)  \\
    F'_{44} &= 0. \numberthis
\end{align*}




%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\newpage{}
\subsection*{Androdioecy via recessive female-sterility mutation}

For the model of the evolution of androdieocy via the invasion of a completely recessive female-sterility mutation at $\mathbf{M}$, only genotypes homozygous for the $M_2$ allele are phenotypically male. Again, the adult genotypic frequencies, $F^A_{ij}$, are described by Equations \ref{eq:AdultFreq} and \ref{eq:Dbar}. Now, the fitness expressions associated with each $F^A_{ij}$ through the female and male sex-function are:

\begin{table}[ht!]
\caption{Fitness expressions for diploid adults prior to reproduction for the model of a recessive female-sterility mutation ($w^f_{ij}$ denotes fitness effects through the female sex-function, $w^m_{ij}$ for male sex-function).}
\centering
\begin{tabular}{l c c c c} \hline
Haplotype & $ AM_1$ & $ AM_2$ & $ aM_1$ & $ aM_2$ \\
\hline
Female-function & & & & \\
$ AM_1$ & $1$ & $1$ & $(1 - h_f s_f)$ & $(1 - h_f s_f)$ \\
$ AM_2$ & $-$ & $0$ & $(1 - h_f s_f)$ & $0$             \\
$ aM_1$ & $-$ & $-$ & $(1 - s_f)$     & $(1 - s_f)$     \\
$ aM_2$ & $-$ & $-$ & $-$             & $0$             \\
Male-function & & & & \\
$ AM_1$ & $(1 - s_m)$ & $(1 - s_m)$        & $(1 - h_m s_m)$ & $(1 - h_m s_m)$        \\
$ AM_2$ & $-$         & $(1 - s_m)(1 + k)$ & $(1 - h_m s_m)$ & $(1 - h_m s_m)(1 + k)$ \\
$ aM_1$ & $-$         & $-$                & $1$             & $1$                    \\
$ aM_2$ & $-$         & $-$                & $-$             & $(1 + k)$              \\
\hline
\end{tabular}
\bigskip{} \\
{\footnotesize Note: Rows and columns indicate the \textit{i}th and \textit{j}th gametic haplotype respectively. The lower triangle of each matrix is the reflection of the upper triangle, and is omitted for simplicity and consistency with the $i \geq j$ row/column indexing used throughout the article.}
\end{table}


\noindent Only individuals not homozygous for $M_2$ are phenotypically hermaphrodite, and thus able to produce ovules. The total number of ovules produced by the population at reproduction is therefore proportional to
\begin{equation} 
O_T = F^A_{11} w^f_{11} + F^A_{12} w^f_{12} + F^A_{13} w^f_{13} + F^A_{14} w^f_{14} + F^A_{23} w^f_{23} + F^A_{33} w^f_{33} + F^A_{34} w^f_{34}.
\end{equation}

\noindent The proportion of ovules that are self-fertilized will therefore be proportional to
\begin{equation} 
S = \frac{C(O_T)}{O_T},
\end{equation}

\noindent and the proportion of ovules fertilized via outcrossing is the complement of $S$:
\begin{equation} 
(1 - S) = \frac{(1 - C)O_T}{O_T}.
\end{equation}

As before, the structure of the equations describing genotypic frequency change among selfed offspring for the model of androdioecy via a recessive female-sterility mutation are similar to the model of gynodioecy via a recessive male-sterility mutation. The proportional contribution of self-fertilized offspring by each of the phenotypically hermaphrodite genotypes, $o^S_{ij}$, follows Eq(\ref{eq:RecSelfOvules}), and the genotypic recursions for self-fertilized zygotes, $G'_{ij}$ follow Eq(\ref{eq:RecSelfRecs}).

The genotypic frequency recursions for offpsring derived via outcrossing proceeds as follows. The proportional contribution of each genotype to the pool of outcrossed ovules are
\begin{align*}
    o^X_{11} &= (1 - C) (F^A_{11} w^f_{11})/O^X_T \\
    o^X_{12} &= (1 - C) (F^A_{12} w^f_{12})/O^X_T \\
    o^X_{13} &= (1 - C) (F^A_{13} w^f_{13})/O^X_T \\
    o^X_{14} &= (1 - C) (F^A_{14} w^f_{14})/O^X_T \\
    o^X_{23} &= (1 - C) (F^A_{23} w^f_{23})/O^X_T \\
    o^X_{33} &= (1 - C) (F^A_{33} w^f_{33})/O^X_T \\
    o^X_{34} &= (1 - C) (F^A_{34} w^f_{34})/O^X_T, \numberthis
\end{align*}

\noindent where $o^X_{ij} = 0$ where $ij \in [22,24,44]$, and $O^X_T = (1 - S)O_T$. The proportional contribution of each genotype that is phenotypically hermaphrodite to the pool of outcrossed pollen is equal to the relative genotypic frequencies after selection:
\begin{align*}
    p^X_{11} &= (F^A_{11} w^f_{11}) / P^X_T \\
    p^X_{12} &= (F^A_{12} w^f_{12}) / P^X_T \\
    p^X_{13} &= (F^A_{13} w^f_{13}) / P^X_T \\
    p^X_{14} &= (F^A_{14} w^f_{14}) / P^X_T \\
    p^X_{22} &= (F^A_{22} w^f_{22}) / P^X_T \\
    p^X_{23} &= (F^A_{23} w^f_{23}) / P^X_T \\
    p^X_{22} &= (F^A_{24} w^f_{24}) / P^X_T \\
    p^X_{34} &= (F^A_{34} w^f_{34}) / P^X_T \\
    p^X_{44} &= (F^A_{44} w^f_{44}) / P^X_T , \numberthis
\end{align*}

\noindent where $P^X_T = \sum_{i=1}^{4}\sum_{j=i}^{4} p^X_{ij}$. The haplotype frequencies among outcrossed ovules ($x_i$) are then:
\begin{align*} 
    x_{1} &= o^X_{11} + \frac{o^X_{12} + o^X_{13} + o^X_{14}}{2} - r \bigg( \frac{o^X_{14} - o^X_{23}}{2} \bigg)  \\
    x_{2} &= \frac{o^X_{12} o^X_{23}}{2} + r \bigg( \frac{o^X_{14} - o^X_{23}}{2} \bigg) \\
    x_{3} &= o^X_{33} + \frac{o^X_{13} + o^X_{23} + o^X_{34}}{2} + r \bigg( \frac{o^X_{14} - o^X_{23}}{2} \bigg)  \\
    x_{4} &= \frac{o^X_{14} o^X_{34}}{2} - r \bigg( \frac{o^X_{14} - o^X_{23}}{2} \bigg).  \numberthis
\end{align*}

\noindent and the haplotype frequencies among the pool of outcrossed pollen ($y_i$) are:
\begin{align*} 
    y_{1} &= p^X_{11} + \frac{p^X_{12} + p^X_{13} + p^X_{14}}{2} - r \bigg( \frac{p^X_{14} - p^X_{23}}{2} \bigg)  \\
    y_{2} &= p^X_{22} + \frac{p^X_{12} + p^X_{23} + p^X_{24}}{2} + r \bigg( \frac{p^X_{14} - p^X_{23}}{2} \bigg)  \\
    y_{3} &= p^X_{33} + \frac{p^X_{13} + p^X_{23} + p^X_{34}}{2} + r \bigg( \frac{p^X_{14} - p^X_{23}}{2} \bigg)  \\
    y_{4} &= p^X_{44} + \frac{p^X_{14} + p^X_{24} + p^X_{34}}{2} - r \bigg( \frac{p^X_{14} - p^X_{23}}{2} \bigg),  \numberthis
\end{align*}

\noindent The genotypic recursions describing the frequency in the next generation of each genotype among zygotes derived from outcrossing are calculated from these haplotype frequences following Eq(\ref{eq:GynRecOutRecs}).

\newpage{}




%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\section*{Appendix B: Expressions for single-locus SA polymorphism}
\renewcommand{\theequation}{B\arabic{equation}}
\setcounter{equation}{0}
\renewcommand{\thefigure}{B\arabic{figure}}
\setcounter{figure}{0}

For the case of obligate outcrossing ($C = 0$) and additive fitness effects ($h_f, h_m = 1/2$) at the SA locus ($\mathbf{A}$), the condition for single-locus polymorphism for an SA locus is:

\begin{equation} 
    \frac{s_m}{1 + s_m} < s_f < \frac{s_m}{1 - s_m}.
\end{equation}

\noindent \citep{Kidwell1977}. The equilibrium frequency of the female-beneficial/male deleterious allele ($A$) in the population is $\hat{p} = (\hat{p}_{f} + \hat{p}_{m})/2$, where $\hat{p}_{m}$ and $\hat{p}_{m}$ are the exact equilibrium frequencies in males and females provided by \citet{Kidwell1977}:

\begin{subequations}
    \begin{align} 
        \hat{p}_{m} &= 1 - \frac{1}{s_f} + \sqrt{\frac{2 + s_f (s_m - 1) - s_m}{2 s_m s_f}} \\
        \hat{p}_{f} &= \frac{1}{s_m} + \sqrt{\frac{2 + s_f (s_m - 1) - s_m}{2 s_m s_f}} 
    \end{align}
\end{subequations}

Using these equilibrium expressions, we calculated the single-locus equilibrium genotypic frequencies for the SA locus as:

\begin{subequations}
    \begin{align} 
        \hat{F}_{AA} &= \hat{p}_f \hat{p}_m \label{eq:obOutGenoEq1} \\
        \hat{F}_{Aa} &= \hat{p}_f (1 - \hat{p}_m) + \hat{p}_m(1 - \hat{p}_f) \label{eq:obOutGenoEq2} \\
        \hat{F}_{aa} &= (1 - \hat{p}_f) (1 - \hat{p}_m) \label{eq:obOutGenoEq3}
    \end{align}
\end{subequations}

Under weak selection ($s_f,s_m \ll 1$), the conditions for polymorphism of SA alleles with additive fitness effects, arbitrary inbreeding depression ($0 < \delta < 1$), and partial selfing ($0 < C < 1$) can be approximated as:

\begin{equation} 
    \frac{s_m(1 - C)}{(s_m - 1)(C(2 \delta - 1) - 1)} < s_f < \frac{s_m(1 - C)}{1 + C + s_m(1 - C) - 2 C \delta}
\end{equation}

\noindent \citep{JordanConnallon2014}. The single-locus equilibrium frequency of the male-beneficial/female-deleterious allele ($a$) with additive fitness effects is:

\begin{equation} 
    \hat{q} = \frac{s_f + s_m(s_f -1) + C(s_m + s_f(1 - s_m - 2 \delta))}{2 s_f(s_m - s_m C \delta)}
\end{equation}

\noindent \citep{JordanConnallon2014}. As described in previous studies \citep{JordanConnallon2014,Olito2016}, when selection is weak ($s_f, s_m \ll 1$), the rate of allele frequency change due to selection will be slow relative to the rate at which genotypic frequencies converge to equilibrium under non-random mating \citep{Nagylaki1997}. In this case, a separation of timescales can be used \citep{OttoDay2007} to calculate quasi-equilibrium (QE) genotypic frequencies which neglect selection. Under partial selfing, the single locus QE genotypic frequencies are:

\begin{subequations}
    \begin{align} 
        F^{\ast}_{AA} &= q^2 + \frac{C q (1 - q)}{(2 - C)} \label{eq:partSelfGenoEq1} \\
        F^{\ast}_{Aa} &= 2 q (1 - q) - \frac{2 C q (1 - q)}{(2 - C)}  \label{eq:partSelfGenoEq2}\\
        F^{\ast}_{AA} &= (1 - q)^2 + \frac{C q (1 - q)}{(2 - C) \label{eq:partSelfGenoEq3}}
    \end{align}
\end{subequations}

\noindent (\citealt{CaballeroHill1992}; see \citealt{JordanConnallon2014,Olito2016} for application to models of SA selection).


%%%%%%%%%%%%%%%%%%%%
\subsection*{Dominance reversals at an SA locus}

When SA fitness effects are partially recessive ($h_f,h_m < 1/2$), the conditions for polymorphism for SA alleles become more permissive than for the case of additive fitness. As for the case of partial selfing and additive fitness above, invasion conditions and equilibrium frequencies must be approximated assuming weak selection ($s_f,s_m \ll 1$). With partially recessive SA fitness effects ($h_f = h_m = 1/4$) and obligate outcrossing ($C = 0$), the conditions for SA polymorphism are:

\begin{equation} 
    \frac{s_m}{3 + s_m} < s_f < \frac{3 s_m}{1 - s_m}.
\end{equation}

\noindent Under these same conditions, the equilibrium frequency of the male-beneficial/female-deleterious SA allele ($a$) is:{}

\begin{equation} 
    \hat{q} = \frac{2 s_f}{s_f + s_m} - \frac{1}{2},
\end{equation}

\noindent and the equilibrium genotypic frequencies at $\mathbf{A}$ are given by \cref{eq:obOutGenoEq1,eq:obOutGenoEq2,eq:obOutGenoEq3}. 

With partial selfing ($0 < C < 1$) and arbitrary inbreeding depression ($0 < \delta < 1$), the conditions for SA polymorphism become:

\begin{align*} 
    \frac{s_m(C^2 - 1)}{(1 + C) (C + s_m(C - 1) - 2(C - 3)C \delta - 3)}& < s_f < \\
    &\frac{s_m(C - 3)(C - 1)}{(1 + C) (s_m - 1) (C(2 \delta - 1) - 1)}. \numberthis
\end{align*}

\noindent \citep{JordanConnallon2014}. Under weak selection, the equilibrium frequency of the male-beneficial/female-deleterious allele ($a$) is:

\begin{equation} 
    \hat{q} = \frac{(C(2 \delta - 1))(s_m(1 - C) + s_f(C+2 C \delta - 3) - 1)}{2(C - 1)(s_m(C - 1)+s_f(C(2 \delta - 1) - 1))},
\end{equation}

\noindent and the QE genotypic frequencies can be calculated using \cref{eq:partSelfGenoEq1,eq:partSelfGenoEq2,eq:partSelfGenoEq3} \citep{JordanConnallon2014}.

When SA effects are completely recessive in each sex (complete dominance reversal; $h_f = h_m = 0$), the conditions for polymorphism and equilibrium allele frequencies simplify considerably \citep{ConnallonJordan2016}. With obligate outcrossing ($C = 0$), all combinations of $0 < s_f,s_f < 1$ permit a single-locus SA polymorphism, and the equilbrium frequency of $a$ becomes:

\begin{equation} 
    \hat{q} = \frac{s_f}{s_f + s_m}.
\end{equation}

\noindent With arbitrary inbreeding depression and partial selfing, the conditions for polymorphism become:

\begin{equation} 
    \frac{C s_m(C - 1)}{C(C - 1)(1 + s_m) - 2 C \delta(C - 2) - 2} < s_f < \frac{s_m(C - 2)(C - 1)}{C(s_m - 1)(C(2 \delta - 1) - 1)},
\end{equation}

\noindent the single-locus equilibrium frequency of the $a$ allele is:

\begin{equation} 
    \hat{q} = \frac{C - 2}{2(C - 1)} - \frac{s_m}{s_m(1 - C) + s_f(1 + C(1 - 2 \delta))},
\end{equation}

\noindent and we can once again calculate the QE genotypic frequencies using \cref{eq:partSelfGenoEq1,eq:partSelfGenoEq2,eq:partSelfGenoEq3}.



\newpage{}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\section*{Appendix C: Results for dominance reversals $(h_f,h_m < 1/2)$}
\renewcommand{\theequation}{C\arabic{equation}}
\setcounter{equation}{0}
\renewcommand{\thefigure}{C\arabic{figure}}
\setcounter{figure}{0}

Following previous theory of SA selection we also explored the the representative and biologically interesting scenario of dominance reversal at the $\mathbf{A}$ locus ($h_f,h_m < 1/2$). Dominance reversals are of particular interest because they are predicted by fitness landscape models of conflicting selection \citep{Fry2010, ConnallonClark2014}, and are generally more permissive of SA polymorphism.



%%%%%%%%%%%%%%%%%%%%%%%%
\subsection*{Invasion into monomorphic populations} 

Under a dominance reversal scenario at the SA locus, invasion criteria for SA alleles are more permissive than when SA fitness effects are additive \citep{Kidwell1977, ConnallonClark2012, JordanConnallon2014, Olito2016}. In our models, a dominance reversal at the SA locus expands the conditions for invasion of a $M_2$ sterility allele. For example, consider a model of gynodioecy, with a dominant male-sterility allele and obligate outcrossing in the ancestral hermaphrodite population ($C=0$). In a population fixed for the female-beneficial SA allele, an incomplete dominance reversal ($h_f=h_m=1/4$) expands conditions for the spread of unisexual females (via $AM_2$ haplotype), relative to the single-locus condition for gynodioecy (Eq(3) in the main text), when:

\begin{equation}\label{eq:2LocGynDomRev}
    \frac{4 r}{3 + r} < s_f.
\end{equation}

\noindent which is more permissive than the analogous result for the additive model of SA genetic variation (i.e., Eq(\ref{eq:2LocGynDomRev}) vs. Eq(3) in the main text).

Linkage to a male-sterility locus expands the parameter conditions for invasion of a female-benefit allele (i.e, beyond the single-locus prediction) when:

\begin{equation}\label{eq:2LocGynDomRevSA}
    r < \frac{s_m}{4 + s_m}.
\end{equation}

As for the case of additive SA fitness effects, replacing $s_f$ with $s_m$ in Eq(\ref{eq:2LocGynDomRev}) gives the analogous conditions for the invasion of unisexual males in the model of androdioecy. In the model of androdioecy, linkage to a female-sterility allele renders the conditions for the invasion of a male-benefit SA allele more permissive than the single-locus predictions when 

\begin{equation}\label{eq:2LocAndDomRevSA}
    r < \frac{9 s_f}{4 + s_f}.
\end{equation}

\noindent The conditions described in Eq(\ref{eq:2LocAndDomRevSA}) are quite permissive, expanding the parameter space where a male benefit allele can invade over single-locus predictions, even under free recombination, so long as $s_f$ is greater than $\sim 0.24$. 

Analogous analytic results for the cases of partial dominance reversal at the SA locus and partial selfing ($0 < C < 1$) are possible, but yield complex expressions that are not easily interpreted. However, examination of complete dominance reversals ($h_f = h_m = 0$) provides some additional insight. In this case, conditions for the spread of a male unisexual sterility allele that is linked to a female-benefit SA allele are relaxed even further. The invasion of unisexual females becomes more permissive than the single-locus prediction as long as $r < s_f$, regardless of the population selfing rate. Again, replacing $s_f$ with $s_m$ in Eq(\ref{eq:2LocGynDomRev}) gives the analogous conditions for the invasion of unisexual males in the model of androdioecy.


%%%%%%%%%%%%%%%%%%%%%%%%
\subsection*{Invasion of unisexuals into polymorphic populations}
When there is a complete dominance reversal at the SA locus ($h_f = h_m = 0$), the conditions for SA polymorphism become far more permissive (e.g., under obligate outcrossing, $C = 0$, the entire parameter space yields polymorphic equilibria) and the parameter space favouring the invasion of unisexuals expands. However, the effects of selfing, inbreeding depression, and recombination remain qualitatively similar between the dominance reversal scenario and results assuming additivity of SA alleles (see Appendix B, figs.~B7--B14 in the Online Supporting Information).

%%%%%%%%%%%%%%%%%%%%%%%%
\subsection*{Equilibrium frequencies of unisexuals when sterility alleles are recessive}
Although the conditions permitting the invasion of unisexuals are less restrictive when there is a dominance reversal at a linked SA locus, the effect of linkage on the equilibrium unisexual frequencies is relatively weak (compare fig.~3 in the main text with supplementary fig.~B14). Under tight linkage ($r = 0$) and adequate reproductive compensation ($k > \hat{k}$), the equilibrium frequency of females (gynodioecy) and males (androdioecy) are slightly higher when SA fitness are partially recessive ($h_f = h_m = 1/4$) relative to the additive case. However, under weaker linkage ($r > 0$) and lower reproductive compensation ($k < \hat{k}$), equilibrium unisexual frequencies are slightly lower ($1 - 2 \%$) than predicted under additive SA fitness effects.

\newpage{}


%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\section*{Appendix D: Supplementary figures}
\renewcommand{\theequation}{D\arabic{equation}}
\setcounter{equation}{0}
\renewcommand{\thefigure}{D\arabic{figure}}
\setcounter{figure}{0}



\begin{figure}[htbp]
\centering
\includegraphics[scale=0.75]{./DblMutExpansionKHatFig}
\caption{Effect of linkage between the $\mathbf{A}$ and $\mathbf{M}$ loci on the threshold level of reproductive compensation for successful invasion of unisexual females (panel A) and males (panel B) into hermaphrodite populations. Plots show the decrease in reproductive compensation for the invasion of 'double mutants' (i.e., based on $\lambda_{\mathbf{AM}}$) compared with the single locus invasion criteria (i.e., based on $\lambda_{\mathbf{M}}$; see Eq(3) and Eq(6) for the models of gyno- and androdioecy respectively) for different selection intensities, selfing rates ($C$) and inbreeding depression ($\delta$) as a function of the recombination rate between the two loci. We vary the recombination rate between $0$ and the threshold where the one- and two-locus invasion conditions become equivalent, Eq(4). For the model of gynodioecy, we first solved $\lambda_{\mathbf{AM}} > 1$ for $k$ (denoted $\hat{k}_{\mathbf{AM}}$ for convenience), and then calculated the proportional decrease in reproductive compensation by subtracting $\hat{k}_{\mathbf{AM}}$ from $\hat{k}$ (the right side of Eq(3)) and dividing by the latter for the specificed values of $C$, $\delta$, $r$, and $s_j$ (where $j \in \{f,m\}$). These results confirm the result presented in fig.~1 of the main text, that the parameter space where unisexual females can invade is increased in the two-locus model because linkage reduces $\hat{k}_{\mathbf{AM}}$ relative to $\hat{k}$ across biologically plausible values of $C$ and $\delta$. Because the integral of Eq(6) is undefined when $C = 1$, we restricted the integration to include high, but not complete, selfing rates $C = [0, 0.9]$ for the model of androdioecy; comparisons between the models should therefore be made with caution.}
\label{fig:dblMutKHatFig}
\end{figure}
\newpage{}


\begin{figure}[ht!]
\centering
\includegraphics[scale=0.58]{./FigD1-Gyno-obOut-funnel}
\caption{Invasion of dominant male-sterility mutations into populations with segregating SA variation under obligate outcrossing. Plots show the fraction of parameter conditions maintaining single-locus SA polymorphism (within the range $0 < s_f,s_m \leq 0.5$) where a dominant male-sterility allele at $\mathbf{M}$ can invade populations initially at single-locus equilibrium frequencies for $\mathbf{A}$ with additive fitness effects ($h_f=h_m=1/2$). Results were obtained by evaluating the two relevant candidate leading eigenvalues ($\lambda_{\mathbf{M}}$,$\lambda_{\mathbf{AM}}$) of the Jacobian matrix of the genotype $\times$ transmission mode recursions for populations at the above initial conditions for $1000$ points uniformly distributed throughout the relevant $s_f \times s_m$ paramter space. Blue points indicate parameter sets where $1 - \lambda_{\mathbf{M}} > 0$, and/or $\lambda_{\mathbf{AM}} - 1 > 0$. Solid black lines represent the corresponding single-locus invasion criteria for SA alleles.}
\label{fig:GynObOutFunnel}
\end{figure}
\newpage{}

\begin{figure}[ht!]
\centering
\includegraphics[scale=0.58]{./FigD2-Gyno-C25-d80-funnel}
\caption{Invasion of dominant male-sterility mutations into populations with segregating SA variation under conditions of low selfing ($C = 0.25$) and high inbreeding depression ($\delta = 0.8$). Plots show the fraction of parameter conditions maintaining single-locus SA polymorphism (within the range $0 < s_f,s_m \leq 0.5$) where a dominant male-sterility allele at $\mathbf{M}$ can invade populations initially at single-locus equilibrium frequencies for $\mathbf{A}$ with additive fitness effects ($h_f=h_m=1/2$). Results were obtained by evaluating the two relevant candidate leading eigenvalues ($\lambda_{\mathbf{M}}$,$\lambda_{\mathbf{AM}}$) of the Jacobian matrix of the genotype $\times$ transmission mode recursions for populations at the above initial conditions for $1000$ points uniformly distributed throughout the relevant $s_f \times s_m$ paramter space. Blue points indicate parameter sets where $1 - \lambda_{\mathbf{M}} > 0$, and/or $\lambda_{\mathbf{AM}} - 1 > 0$. Solid black lines represent the corresponding single-locus invasion criteria for SA alleles.}
\label{fig:GynC25d80Funnel}
\end{figure}
\newpage{}

\begin{figure}[ht!]
\centering
\includegraphics[scale=0.58]{./FigD3-Gyno-C75-d20-funnel}
\caption{Invasion of dominant male-sterility mutations into populations with segregating SA variation under conditions of high selfing ($C = 0.75$) and low inbreeding depression ($\delta = 0.2$). Plots show the fraction of parameter conditions maintaining single-locus SA polymorphism (within the range $0 < s_f,s_m \leq 0.5$) where a dominant male-sterility allele at $\mathbf{M}$ can invade populations initially at single-locus equilibrium frequencies for $\mathbf{A}$ with additive fitness effects ($h_f=h_m=1/2$). Results were obtained by evaluating the two relevant candidate leading eigenvalues ($\lambda_{\mathbf{M}}$,$\lambda_{\mathbf{AM}}$) of the Jacobian matrix of the genotype $\times$ transmission mode recursions for populations at the above initial conditions for $1000$ points uniformly distributed throughout the relevant $s_f \times s_m$ paramter space. Blue points indicate parameter sets where $1 - \lambda_{\mathbf{M}} > 0$, and/or $\lambda_{\mathbf{AM}} - 1 > 0$. Solid black lines represent the corresponding single-locus invasion criteria for SA alleles.}
\label{fig:GynC75d20Funnel}
\end{figure}
\newpage{}

\begin{figure}[ht!]
\centering
\includegraphics[scale=0.58]{./FigD4-Andro-obOut-funnel}
\caption{Invasion of dominant female-sterility mutations into populations with segregating SA variation under obligate outcrossing. Plots show the fraction of parameter conditions maintaining single-locus SA polymorphism (within the range $0 < s_f,s_m \leq 0.5$) where a dominant female-sterility allele at $\mathbf{M}$ can invade populations initially at single-locus equilibrium frequencies for $\mathbf{A}$ with additive fitness effects ($h_f=h_m=1/2$). Results were obtained by evaluating the two relevant candidate leading eigenvalues ($\lambda_{\mathbf{M}}$,$\lambda_{\mathbf{AM}}$) of the Jacobian matrix of the genotype $\times$ transmission mode recursions for populations at the above initial conditions for $1000$ points uniformly distributed throughout the relevant $s_f \times s_m$ paramter space. Blue points indicate parameter sets where $1 - \lambda_{\mathbf{M}} > 0$, and/or $\lambda_{\mathbf{AM}} - 1 > 0$. Solid black lines represent the corresponding single-locus invasion criteria for SA alleles.}
\label{fig:AndroObOutFunnel}
\end{figure}
\newpage{}

\begin{figure}[ht!]
\centering
\includegraphics[scale=0.58]{./FigD5-Andro-C25-d80-funnel}
\caption{Invasion of dominant female-sterility mutations into populations with segregating SA variation under conditions of low selfing ($C = 0.25$) and high inbreeding depression ($\delta = 0.8$). Plots show the fraction of parameter conditions maintaining single-locus SA polymorphism (within the range $0 < s_f,s_m \leq 0.5$) where a dominant female-sterility allele at $\mathbf{M}$ can invade populations initially at single-locus equilibrium frequencies for $\mathbf{A}$ with additive fitness effects ($h_f=h_m=1/2$). Results were obtained by evaluating the two relevant candidate leading eigenvalues ($\lambda_{\mathbf{M}}$,$\lambda_{\mathbf{AM}}$) of the Jacobian matrix of the genotype $\times$ transmission mode recursions for populations at the above initial conditions for $1000$ points uniformly distributed throughout the relevant $s_f \times s_m$ paramter space. Blue points indicate parameter sets where $1 - \lambda_{\mathbf{M}} > 0$, and/or $\lambda_{\mathbf{AM}} - 1 > 0$. Solid black lines represent the corresponding single-locus invasion criteria for SA alleles.}
\label{fig:AndC25d80Funnel}
\end{figure}
\newpage{}

\begin{figure}[ht!]
\centering
\includegraphics[scale=0.58]{./FigD6-Andro-C75-d20-funnel}
\caption{Invasion of dominant female-sterility mutations into populations with segregating SA variation under conditions of high selfing ($C = 0.75$) and low inbreeding depression ($\delta = 0.2$). Plots show the fraction of parameter conditions maintaining single-locus SA polymorphism (within the range $0 < s_f,s_m \leq 0.5$) where a dominant female-sterility allele at $\mathbf{M}$ can invade populations initially at single-locus equilibrium frequencies for $\mathbf{A}$ with additive fitness effects ($h_f=h_m=1/2$). Results were obtained by evaluating the two relevant candidate leading eigenvalues ($\lambda_{\mathbf{M}}$,$\lambda_{\mathbf{AM}}$) of the Jacobian matrix of the genotype $\times$ transmission mode recursions for populations at the above initial conditions for $1000$ points uniformly distributed throughout the relevant $s_f \times s_m$ paramter space. Blue points indicate parameter sets where $1 - \lambda_{\mathbf{M}} > 0$, and/or $\lambda_{\mathbf{AM}} - 1 > 0$. Solid black lines represent the corresponding single-locus invasion criteria for SA alleles.}
\label{fig:AndC75d20Funnel}
\end{figure}
\newpage{}



\begin{figure}[htbp]
\centering
\includegraphics[scale=0.7]{./FigD7R1}
\caption{Invasion of unisexuals into populations with pre-existing SA polymorphism. Plots show the fraction of parameter conditions maintaining single-locus SA polymorphism (within the range $0 < s_f,s_m \leq 0.5$) where a dominant sex-specific sterility allele at $\mathbf{M}$ can invade, assuming completely recessive SA fitness effects at the SA $\mathbf{A}$ locus ($h_f=h_m=0$), plotted as a function of the recombination rate $r$. Panels A--C show results from the model of gynodioecy via invasion of a male-sterility allele, while planels D--F show results for the model of androdioecy via invasion of a female-sterility allele. For each panel, results are shown for different values of reproductive compensation, $k$, chosen as a fraction of the single-locus invasion threshold for $M_2$ ($\hat{k}$, which is equal to the right-hand side of Eq(3) or Eq(6) in the main text for the models of gynodioecy and androdioecy respectively). Hence, the orange, green, and dark blue lines show scenarios where unisexuals experience a decrease in gamete production relative to hermaphrodites of $1$, $5$, and $10\%$. Results were obtained by evaluating the two relevant candidate leading eigenvalues ($\lambda_{\mathbf{A}}$,$\lambda_{\mathbf{M}}$,$\lambda_{\mathbf{AM}}$) of the Jacobian matrix of the genotype $\times$ transmission mode recursions for populations at the above initial conditions for $1000$ points uniformly distributed throughout the relevant $s_f \times s_m$ parameter space.}
\label{fig:PrInvDomRev}
\end{figure}

\begin{figure}[ht!]
\centering
\includegraphics[scale=0.58]{./FigD8-Gyno-domRev-obOut-funnel}
\caption{Invasion of dominant male-sterility mutations into populations with segregating SA variation under obligate outcrossing ($C=0$) and a complete dominance reversal. Plots show the fraction of parameter conditions maintaining single-locus SA polymorphism (within the range $0 < s_f,s_m \leq 0.5$) where a dominant male-sterility allele at $\mathbf{M}$ can invade populations initially at single-locus equilibrium frequencies for $\mathbf{A}$ with completely recessive fitness effects ($h_f=h_m=0$). Results were obtained by evaluating the two relevant candidate leading eigenvalues ($\lambda_{\mathbf{M}}$,$\lambda_{\mathbf{AM}}$) of the Jacobian matrix of the genotype $\times$ transmission mode recursions for populations at the above initial conditions for $1000$ points uniformly distributed throughout the relevant $s_f \times s_m$ paramter space. Blue points indicate parameter sets where $1 - \lambda_{\mathbf{M}} > 0$, and/or $\lambda_{\mathbf{AM}} - 1 > 0$. Solid black lines represent the corresponding single-locus invasion criteria for SA alleles.}
\label{fig:GynObOutFunnelDomRev}
\end{figure}
\newpage{}

\begin{figure}[ht!]
\centering
\includegraphics[scale=0.58]{./FigD9-Gyno-domRev-C25-d80-funnel}
\caption{Invasion of dominant male-sterility mutations into populations with segregating SA variation under conditions of low selfing ($C = 0.25$), high inbreeding depression ($\delta = 0.8$), and a complete dominance reversal. Plots show the fraction of parameter conditions maintaining single-locus SA polymorphism (within the range $0 < s_f,s_m \leq 0.5$) where a dominant male-sterility allele at $\mathbf{M}$ can invade populations initially at single-locus equilibrium frequencies for $\mathbf{A}$ with completely recessive fitness effects ($h_f=h_m=0$). Results were obtained by evaluating the two relevant candidate leading eigenvalues ($\lambda_{\mathbf{M}}$,$\lambda_{\mathbf{AM}}$) of the Jacobian matrix of the genotype $\times$ transmission mode recursions for populations at the above initial conditions for $1000$ points uniformly distributed throughout the relevant $s_f \times s_m$ paramter space. Blue points indicate parameter sets where $1 - \lambda_{\mathbf{M}} > 0$, and/or $\lambda_{\mathbf{AM}} - 1 > 0$. Solid black lines represent the corresponding single-locus invasion criteria for SA alleles.}
\label{fig:GynC25d80FunnelDomRev}
\end{figure}
\newpage{}

\begin{figure}[ht!]
\centering
\includegraphics[scale=0.58]{./FigD10-Gyno-domRev-C75-d20-funnel}
\caption{Invasion of dominant male-sterility mutations into populations with segregating SA variation under conditions of high selfing ($C = 0.75$), low inbreeding depression ($\delta = 0.2$), and a complete dominance reversal. Plots show the fraction of parameter conditions maintaining single-locus SA polymorphism (within the range $0 < s_f,s_m \leq 0.5$) where a dominant male-sterility allele at $\mathbf{M}$ can invade populations initially at single-locus equilibrium frequencies for $\mathbf{A}$ with completely recessive fitness effects ($h_f=h_m=0$). Results were obtained by evaluating the two relevant candidate leading eigenvalues ($\lambda_{\mathbf{M}}$,$\lambda_{\mathbf{AM}}$) of the Jacobian matrix of the genotype $\times$ transmission mode recursions for populations at the above initial conditions for $1000$ points uniformly distributed throughout the relevant $s_f \times s_m$ paramter space. Blue points indicate parameter sets where $1 - \lambda_{\mathbf{M}} > 0$, and/or $\lambda_{\mathbf{AM}} - 1 > 0$. Solid black lines represent the corresponding single-locus invasion criteria for SA alleles.}
\label{fig:GynC75d20FunnelDomRev}
\end{figure}
\newpage{}

\begin{figure}[ht!]
\centering
\includegraphics[scale=0.58]{./FigD11-Andro-domRev-obOut-funnel}
\caption{Invasion of dominant female-sterility mutations into populations with segregating SA variation under obligate outcrossing ($C=0$) and a complete dominance reversal. Plots show the fraction of parameter conditions maintaining single-locus SA polymorphism (within the range $0 < s_f,s_m \leq 0.5$) where a dominant female-sterility allele at $\mathbf{M}$ can invade populations initially at single-locus equilibrium frequencies for $\mathbf{A}$ with completely recessive fitness effects ($h_f=h_m=0$). Results were obtained by evaluating the two relevant candidate leading eigenvalues ($\lambda_{\mathbf{M}}$,$\lambda_{\mathbf{AM}}$) of the Jacobian matrix of the genotype $\times$ transmission mode recursions for populations at the above initial conditions for $1000$ points uniformly distributed throughout the relevant $s_f \times s_m$ paramter space. Blue points indicate parameter sets where $1 - \lambda_{\mathbf{M}} > 0$, and/or $\lambda_{\mathbf{AM}} - 1 > 0$. Solid black lines represent the corresponding single-locus invasion criteria for SA alleles.}
\label{fig:AndObOutFunnelDomRev}
\end{figure}
\newpage{}

\begin{figure}[ht!]
\centering
\includegraphics[scale=0.58]{./FigD12-Andro-domRev-C25-d80-funnel}
\caption{Invasion of dominant female-sterility mutations into populations with segregating SA variation under conditions of low selfing ($C = 0.25$), high inbreeding depression ($\delta = 0.8$), and a complete dominance reversal. Plots show the fraction of parameter conditions maintaining single-locus SA polymorphism (within the range $0 < s_f,s_m \leq 0.5$) where a dominant female-sterility allele at $\mathbf{M}$ can invade populations initially at single-locus equilibrium frequencies for $\mathbf{A}$ with completely recessive fitness effects ($h_f=h_m=0$). Results were obtained by evaluating the two relevant candidate leading eigenvalues ($\lambda_{\mathbf{M}}$,$\lambda_{\mathbf{AM}}$) of the Jacobian matrix of the genotype $\times$ transmission mode recursions for populations at the above initial conditions for $1000$ points uniformly distributed throughout the relevant $s_f \times s_m$ paramter space. Blue points indicate parameter sets where $1 - \lambda_{\mathbf{M}} > 0$, and/or $\lambda_{\mathbf{AM}} - 1 > 0$. Solid black lines represent the corresponding single-locus invasion criteria for SA alleles.}
\label{fig:AndC25d80FunnelDomRev}
\end{figure}
\newpage{}

\begin{figure}[ht!]
\centering
\includegraphics[scale=0.58]{./FigD13-Andro-domRev-C75-d20-funnel}
\caption{Invasion of dominant female-sterility mutations into populations with segregating SA variation under conditions of high selfing ($C = 0.75$), low inbreeding depression ($\delta = 0.2$), and a complete dominance reversal. Plots show the fraction of parameter conditions maintaining single-locus SA polymorphism (within the range $0 < s_f,s_m \leq 0.5$) where a dominant female-sterility allele at $\mathbf{M}$ can invade populations initially at single-locus equilibrium frequencies for $\mathbf{A}$ with completely recessive fitness effects ($h_f=h_m=0$). Results were obtained by evaluating the two relevant candidate leading eigenvalues ($\lambda_{\mathbf{M}}$,$\lambda_{\mathbf{AM}}$) of the Jacobian matrix of the genotype $\times$ transmission mode recursions for populations at the above initial conditions for $1000$ points uniformly distributed throughout the relevant $s_f \times s_m$ paramter space. Blue points indicate parameter sets where $1 - \lambda_{\mathbf{M}} > 0$, and/or $\lambda_{\mathbf{AM}} - 1 > 0$. Solid black lines represent the corresponding single-locus invasion criteria for SA alleles.}
\label{fig:AndC75d20FunnelDomRev}
\end{figure}
\newpage{}


\begin{figure}[ht!]
\centering
\includegraphics[scale=0.75]{./EqFreqAtKHat}
\caption{Effect of linkage on equilibrium frequencies of unisexual (A) females, and (B) males when reproductive compensation is nearly equal to the single-locus threshold for invasion of unisexual sterility allele (i.e., $k = \hat{k} \times 1.001$; where $\hat{k}$ is equal to the right-hand side of Eq(3) or Eq(6) as appropriate). Results are shown for the models of gyno- and androdioecy via invasion of recessive unisexual sterility alleles, additive fitness effects at $\mathbf{A}$ ($h_f = h_m = 0.5$, using selection coefficients of $s_m = 0.1$ (for the model of gynodioecy) and $s_f = 0.1$ (for the model of androdioecy), and inbreeding depression that follows $\delta = \delta^\ast(1 - C/2)$ (see Models section in the main text). Plots illustrate the increase in equilibrium frequencies of unisexuals predicted by our two-locus models (dashed greyscale lines) relative to the corresponding exact single-locus equilibrium frequencies, which are very small at this level of reproductive compensation (solid black line; $\hat{Z}$ predicted by \citealt{Charlesworth1978a}). Results are shown for four different levels of recombination, highlighting that with weaker linkage, the two-locus predictions converge on those of the single-locus model.}
\label{fig:eqFreq2v1Loc}
\end{figure}
\newpage{}

\begin{figure}[ht!]
\centering
\includegraphics[scale=0.75]{./FigD14-Eq-Unisexual-Frequency-DomRev}
\caption{Equilibrium frequencies of unisexual (A--B) females and (C--D) males across a gradient of reproductive compensation. Results are shown for the models of gyno- and androdioecy via invasion of recessive unisexual sterility alleles, recessive SA fitness effects at $\mathbf{A}$ ($h_f = h_m = 1/4$, using selection coefficients of $s_m = 0.1$ (for the model of gynodioecy) and $s_f = 0.1$ (for the model of androdioecy), and inbreeding depression that decreases linearly with the selfing rate: $\delta = \delta^\ast(1 - C/2)$ (see Models section in the main text). Plots show the equilibrium frequencies of unisexuals predicted by our models for five different levels of reproductive compensation, calculated as a fraction of the single-locus invasion criterion for $M_2$ defined by Eq(3) and Eq(6) in the main text. Note that the single-locus equilibrium frequency of unisexuals always equals $0$ when $k < \hat{k}$. Hence, the lines corresponding to $k < \hat{k}$ illustrate how linkage among SA loci expands the parameter space where unisexual sterility alleles can invade beyond the predictions of the single-locus models.}
\label{fig:eqFreqDomRev}
\end{figure}
\newpage{}

\FloatBarrier

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\section*{Appendix E: Alternative relations between the selfing rate and inbreeding depression}
\renewcommand{\theequation}{E\arabic{equation}}
\setcounter{equation}{0}
\renewcommand{\thefigure}{E\arabic{figure}}
\setcounter{figure}{0}

As explained in the main text, we attempted to account for negative covariance between $C$ and $\delta$ in our deterministic simulations, as might be expected if inbreeding depression is caused by recessive deleterious mutations \citep{Charlesworth1987, Charlesworth2009}. We did this by constraining inbreeding depression to follow a simple linear function of the selfing rate, $\delta = \delta^\ast(1 - C/2)$, where $\delta^\ast$ represents the hypothetical severity of inbreeding depression if selfing were enforced on a completely outcrossing population ($\delta^\ast \in [0,1]$). We then used a value of $\delta^{\ast}$ for our simulations that yielded predicted values for $\delta$ that were roughly consistent with empirical data for partially selfing plant populations \citep{HusbandSchemske1996}. Here we briefly address the consequences of relaxing this assumption. However, we reiterate that our goal is not to accurately model inbreeding depression, but to capture the basic effect of purging deleterious recessives in partially selfing populations. 

In the simplest case, the mutation load due to deleterious recessive mutations at a single locus in a completely selfing population should be roughly half that of a randomly mating outcrossing population ($\mu$ versus $2 \mu$, where $\mu$ is the genome-wide mutation rate to recessive deleterious alleles; \citealt{OhtaCockerham1974}). Moreover, a variety of detailed genetic models of inbreeding depression predict that inbreeding depression should follow a negative decelerating curve as a function of the population selfing rate (see, for example, \citealt{OhtaCockerham1974, LandeSchemske1985, Charlesworth1985, Garcia-Dorado2017, LandePorcher2017}). To capture these main features, first we define a simple function describing the mutation load due to recessive deleterious mutations,

\begin{equation}\label{eq:Load}
    L = \frac{a (1 - C)}{C + a (1 - C)},
\end{equation}

\noindent where $C$ is the population selfing rate, and $a$ is a shape parameter determining the curvature of the line. When $a = 1$, Eq(\ref{eq:Load}) yields a straight line. For $a < 1$ the function is concave up (Fig.~\ref{fig:Cdelta}A), and for $a > 1$ it is concave downward. We then incorporate Eq(\ref{eq:Load}) into a simple expression for inbreeding depression,

\begin{equation}\label{eq:ID}
    \delta = \delta^{\ast} - \delta^{\ast} b (1 - L),
\end{equation}

\noindent where $\delta^\ast$ is defined as above, and $b$ is a shape parameter determining how far $\delta$ will decline under complete selfing ($C = 1$) relative to $\delta^{\ast}$. This flexible function allows us to define an initial level of inbreeding depression under obligate outcrossing ($\delta^\ast$), how far inbreeding depression will decline under complete selfing ($b$), and the curvature of the line connecting these two endpoints ($a$). Given that in a completely selfing population $L$ is expected to be roughly half that in an obligately outcrossing one, we constrain $b$ to equal $1/2$. Note that when $a=1$ and $b=1/2$, Eq(\ref{eq:ID}) yields $\delta = \delta^\ast(1 - C/2)$, as defined in the main text. To illustrate the effect of altering the shape of Eq(\ref{eq:Load}) to be concave upward rather than linear, we examine the equilibrium female frequencies predicted by the model of gynodioecy via invasion of a recessive male-sterility allele when reproductive compensation exceeds the single locus threshold for invasion ($k > \hat{k}$). A non-linear relation for $\delta$ has only a minor effect on the equilibrium female frequencies, and does not alter the main conclusions from the simulations (fig.~\ref{fig:Cdelta}B).

\begin{figure}[ht!]
\centering
\includegraphics[width=\linewidth]{FigC1-2-compareCdelta}
\caption{Comparison of linear versus non-linear relations between the hermaphrodite selfing rate, $C$, and inbreeding depression, $\delta$. Panel (A) shows the linear ($a=1$; solid line) vs.~non-linear ($a=0.2$; dashed line) functions for $\delta$. In both cases, we assume that inbreeding depression in a completely selfing population will be half that of an obligately outcrossing one ($b=0.5$ for both lines). Panel (B) shows similar simulation results to those presented in Fig.~2 of the main text -- equilibrium frequencies of unisexual females compared with single-locus predictions when reproductive compensation is above the single-locus threshold (i.e., $k > \hat{k}$). Results are shown for the model of gynodioecy via invasion of a recessive male-sterility allele, with additive fitness effects at $\mathbf{A}$ ($h_f = h_m = 0.5$), using a selection coefficient of $s_m = 0.4$., and the same linear (solid lines) vs.~nonlinear (dashed lines) functions of inbreeding depression as shown in panel (A).}
\label{fig:Cdelta}
\end{figure}



\clearpage

%%%%%%%%%%%%%%%%%%%%%
% Bibliography
%%%%%%%%%%%%%%%%%%%%%
\bibliography{dioecySA-Supplements-bibliography}



\end{document}
