% Preamble
\documentclass{article}

%Dependencies
\usepackage[left]{lineno}
%\usepackage{indentfirst}
\usepackage{titlesec}
\usepackage{amsmath}
\usepackage{amsfonts}
\usepackage{amssymb}
\usepackage[utf8]{inputenc}
\usepackage{amsmath}
\usepackage{amsfonts}
\usepackage{amssymb}
\usepackage{color,soul}
%\usepackage{times}
\usepackage[sc]{mathpazo} %Like Palatino with extensive math support
\usepackage[authoryear,sectionbib,sort]{natbib}
\usepackage[hidelinks]{hyperref}
\linespread{1.25}
% Default margins are too wide all the way around. I reset them here
\usepackage{fullpage}

% Running headers
\usepackage{fancyhdr}
\setlength{\headheight}{28pt}

% Graphics package
\usepackage{graphicx}
\graphicspath{{../output/figures/}.pdf}

% New commands: fonts
%\newcommand{\code}{\fontfamily{pcr}\selectfont}
%\newcommand*\chem[1]{\ensuremath{\mathrm{#1}}}
\newcommand\numberthis{\addtocounter{equation}{1}\tag{\theequation}}

%% Other Options

% Change subsection numbering
\renewcommand\thesubsection{\arabic{subsection})}
\renewcommand\thesubsubsection{}

% Subsubsection Title Formatting
\titleformat{\subsubsection}    
{\normalfont\fontsize{12pt}{17}\itshape}{\thesubsubsection}{12pt}{}


%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\title{Supplementary Material (Appendices A -- \hl{X}) for: Sexually antagonistic polymorphism and the evolution of dimporphic sexual systems in hermaphrodites. \textit{Genetics}}

\author{Colin Olito \& Tim Connallon}

\date{\today}

\begin{document}
\maketitle

\noindent{} School of Biological Sciences, Monash University, Melbourne, VIC 3800, Australia.
\bigskip

\noindent{} Corresponding author e-mail: \url{colin.olito@gmail.com}
\bigskip

\newpage
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
% Running Header
%\pagestyle{fancyplain}
%\makeatother
%\lhead{\textit{Supplement to Olito \& Connallon. SA polymorphism and dimorphic sexual systems. \textit{Evolution}.\\}}
%\rhead{\textit{Dimorphic sexual systems}}
%\renewcommand{\headrulewidth}{0pt}
%\renewcommand{\footrulewidth}{0pt}
%\addtolength{\headheight}{12pt}


\subsection*{Appendix A: Development of the recursions -- }
\renewcommand{\theequation}{A\arabic{equation}}
\titleformat{\subsubsection}    
{\normalfont\fontsize{12pt}{17}\itshape}{\thesubsubsection}{12pt}{}



\subsubsection*{Recursions}

Consider the genetic system described in the main text. Adult genotypic frequencies formed by the union of the $i$th and $j$th haplotypes are denoted using $F_{ij}$, and the overall haplotype frequencies, $q_i$ are

\begin{align*}
	q_1 &= F_{11} + \frac{F_{12} + F_{13} + F_{14}}{2} \\
	q_2 &= F_{22} + \frac{F_{12} + F_{23} + F_{24}}{2} \\
	q_3 &= F_{33} + \frac{F_{13} + F_{23} + F_{34}}{2} \\
	q_4 &= F_{44} + \frac{F_{14} + F_{24} + F_{34}}{2}. \numberthis \label{eqn}
\end{align*} 

Haplotype frequencies in female gametes are denoted $x_1 = [AB]$, $x_2 = [Ab]$, $x_3 = [aB]$, and $x_4 = [ab]$, and the corresponding frequencies in male gametes are denoted $y_1$, $y_2$, $y_3$, and $y_4$. The contribution of genotype $ij$ to the production of offspring in the next generation is denoted by $F^f_{ij}$. Given these conditions, the recursion equations for the genotypic frequencies in the next generation are (\citealt{Holden1979, JordanConn2014})

\begin{align*} \label{eq:genRec}
	F'_{11} &= (1 - C) (x_1 y_1)           + C \bigg( F^f_{11} + \frac{F^f_{12} + F^f_{13} + F^f_{14}(1 - r)^2 + F^f_{23} r^2}{4} \bigg)  \\
	F'_{12} &= (1 - C) (x_1 y_2 + x_2 y_1) + C \bigg( \frac{F^f_{12} + F^f_{14} r(1-r) + F^f_{23} r(1-r)}{2} \bigg)  \\
	F'_{13} &= (1 - C) (x_1 y_3 + x_3 y_1) + C \bigg( \frac{F^f_{13} + F^f_{14} r(1-r) + F^f_{23} r(1-r)}{2} \bigg)  \\
	F'_{14} &= (1 - C) (x_1 y_4 + x_4 y_1) + C \bigg( \frac{F^f_{14} (1-r)^2 + F^f_{23} r^2}{2} \bigg)  \\
	F'_{22} &= (1 - C) (x_2 y_2)           + C \bigg( F^f_{22} + \frac{F^f_{12} + F^f_{14} r^2 + F^f_{23} (1-r)^2 + F^f_{24}}{4} \bigg)  \\
	F'_{23} &= (1 - C) (x_2 y_3 + x_3 y_2) + C \bigg( \frac{F^f_{14} r^2 + F^f_{23} (1-r)^2}{2} \bigg)  \\
	F'_{24} &= (1 - C) (x_2 y_4 + x_4 y_2) + C \bigg( \frac{F^f_{24} + F^f_{14} r(1-r) + F^f_{23} r(1-r)}{2} \bigg)  \\
	F'_{33} &= (1 - C) (x_3 y_3)           + C \bigg( F^f_{33} + \frac{F^f_{13} + F^f_{14} r^2 + F^f_{23} (1-r)^2 + F^f_{34}}{4} \bigg)  \\
	F'_{34} &= (1 - C) (x_3 y_4 + x_4 y_3) + C \bigg( \frac{F^f_{14} r(1-r) + F^f_{23} r(1-r) + F^f_{34}}{2} \bigg)  \\
	F'_{44} &= (1 - C) (x_4 y_4)           + C \bigg( F^f_{44} + \frac{F^f_{14} (1-r)^2 + F^f_{23} r^2 + F^f_{24} + F^f_{34}}{4} \bigg)  \numberthis
\end{align*}

\noindent{} where $x_{i}$ are

\begin{align*}
	x_1 &= \frac{2 F_{11} w_{f11} + F_{12} w_{f12} + F_{13} w_{f13} + F_{14} w_{f14}}{2 \overline{w}_f} - r \bigg( \frac{F_{14} w_{f14} - F_{23} w_{f23}}{2 \overline{w}_f} \bigg) \\
	x_2 &= \frac{2 F_{22} w_{f22} + F_{12} w_{f12} + F_{23} w_{f23} + F_{24} w_{f24}}{2 \overline{w}_f} + r \bigg( \frac{F_{14} w_{f14} - F_{23} w_{f23}}{2 \overline{w}_f} \bigg) \\
	x_3 &= \frac{2 F_{33} w_{f33} + F_{34} w_{f34} + F_{13} w_{f13} + F_{23} w_{f23}}{2 \overline{w}_f} + r \bigg( \frac{F_{14} w_{f14} - F_{23} w_{f23}}{2 \overline{w}_f} \bigg) \\
	x_4 &= \frac{2 F_{44} w_{f44} + F_{32} w_{f32} + F_{14} w_{f14} + F_{24} w_{f24}}{2 \overline{w}_f} - r \bigg( \frac{F_{14} w_{f14} - F_{23} w_{f23}}{2 \overline{w}_f} \bigg), \numberthis
\end{align*}

\noindent{} $y_{i}$ are

\begin{align*}
	y_1 &= \frac{2 F_{11} w_{m11} + F_{12} w_{m12} + F_{13} w_{m13} + F_{14} w_{m14}}{2 \overline{w}_m} - r \bigg( \frac{F_{14} w_{m14} - F_{23} w_{m14}}{2 \overline{w}_m} \bigg) \\
	y_2 &= \frac{2 F_{22} w_{m22} + F_{12} w_{m12} + F_{23} w_{m23} + F_{24} w_{m24}}{2 \overline{w}_m} + r \bigg( \frac{F_{14} w_{m14} - F_{23} w_{m23}}{2 \overline{w}_m} \bigg) \\
	y_3 &= \frac{2 F_{33} w_{m33} + F_{34} w_{m34} + F_{13} w_{m13} + F_{23} w_{m23}}{2 \overline{w}_m} + r \bigg( \frac{F_{14} w_{m14} - F_{23} w_{m23}}{2 \overline{w}_m} \bigg) \\
	y_4 &= \frac{2 F_{44} w_{m44} + F_{32} w_{m32} + F_{14} w_{m14} + F_{24} w_{m24}}{2 \overline{w}_m} - r \bigg( \frac{F_{14} w_{m14} - F_{23} w_{m23}}{2 \overline{w}_m} \bigg), \numberthis
\end{align*}

\noindent{} where ${w}_{k,ij}$ represent the fitness through each sex function ($k \in [m,f]$) of adults resulting from the union of the $i$th and $j$th haplotypes, and the population mean fitness through each sex function is

\begin{align*}
	\overline{w}_f = F_{11} w_{f11} + &F_{12} w_{f12} + F_{13} w_{f13} + F_{14} w_{f14} + F_{22} w_{f22} + \\ 
				     &F_{23} w_{f23} + F_{24} w_{f24} + F_{33} w_{f33} + F_{34} w_{f34} + F_{44} w_{f44} \numberthis \\
	\overline{w}_m = F_{11} w_{m11} + &F_{12} w_{m12} + F_{13} w_{m13} + F_{14} w_{m14} + F_{22} w_{m22} + \\
				     &F_{23} w_{m23} + F_{24} w_{m24} + F_{33} w_{m33} + F_{34} w_{m34} + F_{44} w_{m44}. \numberthis
\end{align*}


Combining Eqns(A1-A6) and simplifying yields the general haplotype recursion equations: 

\begin{align*} \label{eq:hapRec}
	q'_1 &= F'_{11} + \frac{F'_{12} + F'_{13} + F'_{14}}{2} = (1 - C) \frac{(x_1 + y_1)}{2} + C \bigg( \frac{x_1 - r(F_{14} - F_{23})}{2 \overline{w}_f} \bigg) \\
	q'_2 &= F'_{22} + \frac{F'_{12} + F'_{23} + F'_{24}}{2} = (1 - C) \frac{(x_2 + y_2)}{2} + C \bigg( \frac{x_2 + r(F_{14} - F_{23})}{2 \overline{w}_f} \bigg) \\
	q'_3 &= F'_{33} + \frac{F'_{13} + F'_{23} + F'_{34}}{2} = (1 - C) \frac{(x_3 + y_3)}{2} + C \bigg( \frac{x_3 + r(F_{14} - F_{23})}{2 \overline{w}_f} \bigg) \\
	q'_4 &= F'_{44} + \frac{F'_{14} + F'_{24} + F'_{34}}{2} = (1 - C) \frac{(x_4 + y_4)}{2} + C \bigg( \frac{x_4 - r(F_{14} - F_{23})}{2 \overline{w}_f} \bigg). \numberthis
\end{align*}

\subsubsection*{QE Approximations}

After \citet{CaballeroHill1992}, the QE genotypic frequencies for a single locus, $\mathbf{A}$, are

\begin{align*}
	F_{AA}^* &= q^2 + \frac{C q(1 - q)}{(2 - C)} \\
	F_{Aa}^* &= 2 q (1 - q) - \frac{2 C q(1 - q)}{(2 - C)} \\
	F_{aa}^* &= (1 - q)^2 + \frac{C q(1 - q)}{2 - C}, \numberthis 
\end{align*}

\noindent{} where $q$ is the allele frequency and $C$ is the fixed proportion of self fertilization. We can extend these single-locus results for the two-locus model to calculate the QE adult genotypic frequencies formed by the union of the $i$th and $j$th haplotypes:

\begin{align*}
	\phi_{11} &= q_1^2 + C q_1 (1 - q_1) / (2-C)  \\
	\phi_{12} &= 4 q_1 q_2 (1 - C) / (2-C)        \\
	\phi_{13} &= 4 q_1 q_3 (1 - C) / (2-C)        \\
	\phi_{14} &= 4 q_1 q_4 (1 - C) / (2-C)        \\
	\phi_{22} &= q_2^2 + C q_2 (1 - q_2) / (2-C)  \\
	\phi_{23} &= 4 q_2 q_3 (1 - C) / (2-C)        \\
	\phi_{24} &= 4 q_2 q_4 (1 - C) / (2-C)        \\
	\phi_{33} &= q_3^2 + C q_3 (1 - q_3) / (2-C)  \\
	\phi_{34} &= 4 q_3 q_4 (1 - C) / (2-C)        \\
	\phi_{44} &= q_4^2 + C q_4 (1 - q_4) / (2-C)  \numberthis
\end{align*}

Following standard two-locus theory for partially selfing populations (\citealt{Holden1979, OttoDay2007, JordanConn2014}), and substituting the QE genotypic frequencies, $\phi_{ij}$, yields the haplotype recursions given in Eq(1), where $x_{i}$ are

\begin{align*}
	x_1 &= \frac{2 \phi_{11} w_{f11} + \phi_{12} w_{f12} + \phi_{13} w_{f13} + \phi_{14} w_{f14}}{2 \overline{w}_f} -
				r \bigg( \frac{\phi_{14} w_{f14} - \phi_{23} w_{f23}} {2 \overline{w}_f} \bigg) \\
	x_2 &= \frac{2 \phi_{22} w_{f22} + \phi_{12} w_{f12} + \phi_{23} w_{f23} + \phi_{24} w_{f24}}{2 \overline{w}_f} +
				r \bigg( \frac{\phi_{14} w_{f14} - \phi_{23} w_{f23}} {2 \overline{w}_f} \bigg) \\ 
	x_3 &= \frac{2 \phi_{33} w_{f33} + \phi_{34} w_{f34} + \phi_{13} w_{f13} + \phi_{23} w_{f23}}{2 \overline{w}_f} +
				r \bigg( \frac{\phi_{14} w_{f14} - \phi_{23} w_{f23}} {2 \overline{w}_f} \bigg) \\
	x_4 &= \frac{2 \phi_{44} w_{f44} + \phi_{34} w_{f34} + \phi_{13} w_{f13} + \phi_{24} w_{f24}}{2 \overline{w}_f} -
				r \bigg( \frac{\phi_{14} w_{f14} - \phi_{23} w_{f23}} {2 \overline{w}_f} \bigg) \numberthis
\end{align*}

\noindent{} and $y_{i}$ are

\begin{align*}
	y_1 &= \frac{2 \phi_{11} w_{m11} + \phi_{12} w_{m12} + \phi_{13} w_{m13} + \phi_{14} w_{m14}}{2 \overline{w}_m} -
				r \bigg( \frac{\phi_{14} w_{m14} - \phi_{23} w_{m23}} {2 \overline{w}_m} \bigg) \\
	y_2 &= \frac{2 \phi_{22} w_{m22} + \phi_{12} w_{m12} + \phi_{23} w_{m23} + \phi_{24} w_{m24}}{2 \overline{w}_m} +
				r \bigg( \frac{\phi_{14} w_{m14} - \phi_{23} w_{m23}} {2 \overline{w}_m} \bigg) \\
	y_3 &= \frac{2 \phi_{33} w_{m33} + \phi_{34} w_{m34} + \phi_{13} w_{m13} + \phi_{23} w_{m23}}{2 \overline{w}_m} +
				r \bigg( \frac{\phi_{14} w_{m14} - \phi_{23} w_{m23}} {2 \overline{w}_m} \bigg) \\
	y_4 &= \frac{2 \phi_{44} w_{m44} + \phi_{34} w_{m34} + \phi_{13} w_{m13} + \phi_{24} w_{m24}}{2 \overline{w}_m} -
				r \bigg( \frac{\phi_{14} w_{m14} - \phi_{23} w_{m23}} {2 \overline{w}_m} \bigg) \numberthis
\end{align*}

\noindent{} and the population mean fitness through each sex function is

\begin{align*}
	\overline{w}_f = \phi_{11} w_{f11} + &\phi_{12} w_{f12} + \phi_{13} w_{f13} + \phi_{14} w_{f14} + \phi_{22} w_{f22} + \\ 
				     &\phi_{23} w_{f23} + \phi_{24} w_{f24} + \phi_{33} w_{f33} + \phi_{34} w_{f34} + \phi_{44} w_{f44} \numberthis \\
	\overline{w}_m = \phi_{11} w_{m11} + &\phi_{12} w_{m12} + \phi_{13} w_{m13} + \phi_{14} w_{m14} + \phi_{22} w_{m22} + \\
				     &\phi_{23} w_{m23} + \phi_{24} w_{m24} + \phi_{33} w_{m33} + \phi_{34} w_{m34} + \phi_{44} w_{m44}. \numberthis
\end{align*}



\newpage{}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\subsection*{Appendix B: Supplementary figures}
\renewcommand{\theequation}{B\arabic{equation}}
\setcounter{equation}{0}
\renewcommand{\thefigure}{B\arabic{figure}}
\setcounter{figure}{0}

\begin{figure}[h!]
\includegraphics[scale=0.7]{./recSimFig_add}
\caption{Haplotype recursions evaluated at QE, and the resulting invasion conditions, approximate the evolutionary trajectory of the full genotypic recursions very well under additive allelic effects, even under strong selection. Predicted regions of SA polymorphism based on deterministic simulations using the genotypic recursions Eq(2) are compared against the outcome of the invasion analysis for the haplotype recursions using the QE approximation (Eq 1) across a gradient of selfing ($C$) and recombination ($r$) rates. Green points indicate parameter conditions where deterministic simulations of the genotypic recursions (Sim.) and the invasion analysis based on eigenvalues for the QE haplotype approximations (Eig.) both predicted polymorphism. Red points indicate regions where the Sim.~predicted polymorphism but the Eig.~did not; and blue points indicate the opposite. The proportion of each outcome is shown in the upper left corner of each panel. Black solid lines show the invasion conditions based on the haplotype recursions using the QE approximation for the given values of $C$ and $r$, with lines drawn for both the two-locus and single-locus invasion conditions when $r > 0$.}
\label{fig:addSim}
\end{figure}
\newpage{}


\begin{figure}[h!] 
\includegraphics[scale=0.7]{./recSimFig_domRev}
\caption{Haplotype recursions evaluated at QE, and the resulting invasion conditions, approximate the evolutionary trajectory of the full genotypic recursions very well under allelic effects resulting in dominance reversal, even under strong selection. Predicted regions of SA polymorphism based on deterministic simulations using the genotypic recursions Eq(2) are compared against the outcome of the invasion analysis for the haplotype recursions using the QE approximation (Eq 1) across a gradient of selfing ($C$) and recombination ($r$) rates. Green points indicate parameter conditions where deterministic simulations of the genotypic recursions (Sim.) and the invasion analysis based on eigenvalues for the QE haplotype approximations (Eig.) both predicted polymorphism. Red points indicate regions where the Sim.~predicted polymorphism but the Eig.~did not; and blue points indicate the opposite. The proportion of each outcome is shown in the upper left corner of each panel. Black solid lines show the invasion conditions based on the haplotype recursions using the QE approximation for the given values of $C$ and $r$, with lines drawn for both the two-locus and single-locus invasion conditions when $r > 0$.}
\label{fig:domRevSim}
\end{figure}
\newpage{}


\begin{figure}[h!]
\includegraphics[scale=0.75]{./Fig1wk}
\caption{Under weak selection ($s_f,s_m < 0.1$), the parameter space where sexually antagonistic polymorphism is maintained is only expanded beyond the single-locus case when there is very tight linkage ($r \approx 0$). Thus, there is little scope for self-fertilization to influence sexually antagonistic polymorphism. Results are shown for the conditions of additive allelic effects ($h_i = 1/2$; panels A--C), and dominance reversal ($h_i = 1/4$; panels D--F). In each panel, the shaded region between the black lines indicates the region of sexually antagonistic polymorphism for the single-locus case; grey lines indicate the thresholds for invasion for female-beneficial (lower lines) and male-beneficial (upper lines) alleles under complete linkage ($r = 0$).}
\label{fig:wkFunnelPlots}
\end{figure}
\newpage{}


\begin{figure}[h!]
\includegraphics[scale=0.8]{./Fig2wk}
\caption{Under weak selection, the proportion of parameter space where SA polymorphism is maintained again declines with increased recombination and self-fertilization (panels A,C). The increase in SA polymorphism in the two-locus model relative to the single-locus case also declines with the recombination rate, but increases with the rate of self-fertilization (panels B,D). However, these effects are limited to cases of extremely tight linkage ($r < 0.004$). Results are shown for the conditions of additive allelic effects ($h_i = 1/2$; panels A--B), and dominance reversal ($h_i = 1/4$; panels C--D). Results were obtained by evaluating the leading eigenvalues of $\mathbf{J}$ for populations initially fixed for the haplotypies $[AB]$ and $ab$ at 30,000 points distributed uniformly throughout parameter space defined by $s_m \times s_f$, where $s_f,s_m \in [0,0.1]$}
\label{fig:wkPolymorhism}
\end{figure}

\newpage
%%%%%%%%%%%%%%%%%%%%%
% Bibliography
%%%%%%%%%%%%%%%%%%%%%

\bibliography{dioecySA-Supplements-bibliography}

\end{document}
