\documentclass[9pt,twocolumn,twoside,lineno]{gsajnl}
% Use the documentclass option 'lineno' to view line numbers

% Equation numbering
\newcommand\numberthis{\addtocounter{equation}{1}\tag{\theequation}}

\articletype{inv} % article type
% {inv} Investigation 
% {gs} Genomic Selection
% {goi} Genetics of Immunity 
% {gos} Genetics of Sex 
% {mp} Multiparental Populations

\title{Sexually antagonistic polymorphism and the evolution of dimorphic sexual systems from hermaphroditism}

\author[$\ast$,1]{Colin Olito}
\author[$\ast$]{Tim Connallon}
% \author[$\dagger$]{Author Two}
% \author[$\ddagger$]{Author Three}
% \author[$\S$]{Author Four}
% \author[$\ast\ast$]{Author Five}

\affil[$\ast$]{Department of Biological Sciences, Monash University, Melbourne, VIC 3800, Australia}
% \affil[$\dagger$]{Author two affiliation}
% \affil[$\ddagger$]{Author three affiliation}
% \affil[$\S$]{Author four affiliation}
% \affil[$\ast\ast$]{Author five affiliation}

\keywords{Keyword; Keyword2; Keyword3; ...}

\runningtitle{GENETICS Journal Template on Overleaf} % For use in the footer 

\correspondingauthor{Colin Olito}

\begin{abstract}
The abstract should be written for people who may not read the entire paper, so it must stand on its own. The impression it makes usually determines whether the reader will go on to read the article, so the abstract must be engaging, clear, and concise. In addition, the abstract may be the only part of the article that is indexed in databases, so it must accurately reflect the content of the article. A well-written abstract is the  most effective way to reach intended readers, leading to more robust search, retrieval, and usage of the article. 

Please see additional guidelines notes on preparing your abstract below.
\end{abstract}

\setboolean{displaycopyright}{true}

\begin{document}

\maketitle
\thispagestyle{firststyle}
\marginmark
\firstpagefootnote
\correspondingauthoraffiliation{Department of Biological Sciences, Monash University, Melbourne, VIC 3800, Australia. Email: colin.olito@gmail.com.}
\vspace{-11pt}%

\lettrine[lines=2]{\color{color2}A}{}lthough the majority of plants are hermaphroditic (individuals perform both male and female sex-functions), plant sexual systems form a diverse spectrum ranging from simultaneous hermaphroditism, to dioecy (physically separate sexes), and all possible intermediate states. One major evolutionary pathway from hermaphroditism towards separate sexes is via the sequental invasion of a sex-specific sterility mutations, which ultimately become linked to form a sex-determining-locus. Classic population genetic theory underpinning this evolutionary pathway predicts (1) that for the first sterility mutation to invade a population of hermaphrodites there must be accompanying compensation in sex-function for unisexuals, (2) invasion of the sterility mutation is facilitated by higher self-fertization rates and stronger inbreeding depression, and (3) that invasion of male-sterility mutations is more permissive than for female-sterility mutations.

However, sterility mutations can be viewed as a class of strongly deleterious sexually antagonistic (SA) alleles. Recent theory predicts that linkage and 'genetic architecture' can strongly affect the evolution of SA alleles. Here, we explore the theoretical consequences of physical linkage between a 'standard' SA locus and a second 'sterility' locus for the invasion of sterility mutations into hermaphrodite populations. We show that linkage results in significantly more permissive parameter conditions for the invasion of unisexuals, and elevated equilibrium frequencies of unisexuals, relative to the classic single-locus predictions. Overall, our results suggest that sterility mutations driving the evolution of dimorphic sexual systems may evolve in tight linkage with pre-existing polymorphic SA loci, with potentially interesting implications for the process of early sex-chromosome evolution.



\section{Models}
\label{sec:materials:methods}

Following previous population genetic models of the evolution of dimorphic sexual systems \citep{Charlesworth1978}, we model the evolution of gyno- and androdioecy from an ancestral population of simultaneous hermaphrodites via the invasion of alleles causing complete male- and female-sterility. We restrict our analyses to four scenarios of biological interest: the evolution of (1) gynodioecy via invasion of a completely dominant male-sterility allele; (2) gynodioecy via invasion of a completely recessive male-sterility allele, (3) androdioecy via the invasion of a completely dominant female-sterility allele, and finally (4) androdioecy via the invasion of a completely recessive female-sterility allele. These scenarios represent 'corner cases' for both the phenotypic effect and the dominance of the sterility mutations. Alternative models allowing for partial sterility and/or dominance will likely yield evolutionary dynamics that are intermediate to our results. Here we briefly describe the simplest case (1) upon which we base most of our analytic results, and highlight only essential differences between (1) and the other three models. A full development of all models can be found in Appendix A, and all code necessary to reproduce the analyses are available in the Online Supporting Information, and at \url{https://github.com/colin-olito/dioecySA}.

\subsection{Gynodioecy}
Consider a genetic system involving two diallelic autosomal loci, $\mathbf{A}$ (with alleles $A$, $a$) and $\mathbf{M}$ (with alleles $M_1$, $M_2$), that recombine at rate $r$ in a large population of simultaneous hermaphrodites. Let us assume that the $\mathbf{A}$ locus is under sexually antagonistic selection, with the $A$ allele having female-beneficial (male-deleterious) fitness effects, and the $a$ allele having male-beneficial (female-deleterious) fitness effects. Let us also assume that the $M_1$ allele at the $\mathbf{M}$ locus is neutral, while the $M_2$ allele causes sterility through the male sex function (e.g., via cessation of pollen production, or the production of inviable pollen), and is completely dominant to the $M_1$ allele. In this genetic system, the successful invasion of the $M_2$ allele, either alone or coupled with an allele at the $\mathbf{A}$ locus as part of a haplotype, represents the evolution of gynodioecy from simultaneous hermaphroditism. 

The population rate of self-fertilization, $C$, is assumed to be independent of the genotype at the $\mathbf{A}$ locus. In contrast, the genotype at the $\mathbf{M}$ locus determines whether or not self-fertilization is possible (after \citealt{Charlesworth1978}). We also assume a constant population level of inbreeding depression, $\delta$, defined as the decrease in the probability of survival of zygotes formed by self-fertilization relative to those produced by outcrossing. As noted by \citet{Charlesworth1978}, this assumption probably represents a conservative estimate of the effects of inbreeding depression. This is because inbreeding depression generally increases with the population rate of outcrossing \citep{Charlesworth2009}, and we are exploring the invasion of a sterility allele that increases the rate of outcrossing. Generations are assumed to be non-overlapping, and the life-cycle proceeds as follows: fertilization $\rightarrow$ inbreeding depression $\rightarrow$ selection $\rightarrow$ fertilization.

Let $x_i$ and $y_i$ denote the frequencies of the four possible haplotypes $[AM_1,AM_2,aM_1,aM_2]$ in female and male gametes respectively. Under some circumstances, female unisexuals carring the $M_2$ allele may be able to re-allocate resources towards ovule production that would otherwise have been used for pollen production (\citealt{Lloyd1975,Lloyd1976,Charlesworth1978}). To account for this gametic compensation, let $k$ describe the proportional increase in ovule production in females relative to hermaphrodites. The fitness of offspring resulting from the union of the i$^{th}$ female and j$^{th}$ male gametic haplotype, are denoted $w^f_{ij}$ and $w^m_{ij}$ respectively, and are assumed to be the product of the fitness at $\mathbf{A}$ and $\mathbf{M}$ (Table \ref{tab:fitness}).

\begin{table*}[htbp]
\centering
\caption{\bf Fitness expressions for selection on diploid adults prior to reproduction ($w^f_{ij}$ for females, $w^m_{ij}$ for males).}
\begin{tableminipage}{\textwidth}
\begin{tabularx}{\textwidth}{XXXXX}
\hline
Haplotype & $y_1 = AM_1$ & $y_2 = AM_2$ & $y_3 = aM_1$ & $y_4 = aM_2$ \\
\hline
Female-function & & & & \\
$x_1 = AM_1$ & $1$ & $(1 + k)$ & $(1 - h_f s_f)$        & $(1 - h_f s_f)(1 + k)$ \\
$x_2 = AM_2$ & $-$ & $(1 + k)$ & $(1 - h_f s_f)(1 + k)$ & $(1 - h_f s_f)(1 + k)$ \\
$x_3 = aM_1$ & $-$ & $-$       & $(1 - s_f)$            & $(1 - s_f)(1 + k)$ \\
$x_4 = aM_2$ & $-$ & $-$       & $-$                    & $(1 - s_f)(1 + k)$ \\
Male-function & & & & \\
$x_1 = AM_1$ & $(1 - s_m)$ & $0$ & $(1 - h_m s_m)$ & $0$ \\
$x_2 = AM_2$ & $-$         & $0$ & $0$             & $0$ \\
$x_3 = aM_1$ & $-$         & $-$ & $1$             & $0$ \\
$x_4 = aM_2$ & $-$         & $-$ & $-$             & $0$ \\
\hline
\end{tabularx}
  \label{tab:fitness}
\end{tableminipage}
{\footnotesize Note: Rows and columns indicate the haplotype inherited from mothers and fathers respectively. The lower triangle of each matrix is the reflection of the upper triangle, and is omitted for simplicity and consistency with the $ij$ row/column indexing used throughout the article.}
\end{table*}

Non-independence between an individual's genotype at the $\mathbf{M}$ locus and the ability to self-fertilize complicates the derivation of recursion equations describing evolutionary change in genotype frequencies. We therefore model the change in frequency of each genotype given the mode of transmission (either self-fertilization or outcrossing). This approach yields a system of $20$ recursion equations ($10$ genotypes $\times$ two modes of transmission). However, for the case of completely a dominant $M_2$ allele, the system of recursions reduces considerably. Let $F_{ij}$ equal the frequency among zygotes of the genotype formed by the union of the \textit{i}th female and \textit{j}th male gametic haplotype via outcrossing, and $G_{ij}$ be the same for zygotes formed via self-fertilization. The genotypic frequencies in the next generation among zygotes formed by outcross fertilization are then:
\begin{align*} \label{eq:Fprimes}
    F'_{11} &= (x_1 y_1) (1 - S) \\
    F'_{12} &= (x_2 y_1) (1 - S) \\
    F'_{13} &= (x_1 y_3 + x_3 y_1) (1 - S) \\
    F'_{14} &= (x_4 y_1) (1 - S) \\
    F'_{22} &= 0 \\
    F'_{23} &= (x_2 y_3) (1 - S) \\
    F'_{24} &= 0 \\
    F'_{33} &= (x_3 y_3) (1 - S) \\
    F'_{34} &= (x_4 y_3) (1 - S) \\
    F'_{44} &= 0, \numberthis
\end{align*}

\noindent where $x_{ij}$ and $y_{ij}$ are functions $f(F_{ij},G_{ij},w^f_{ij},w^m_{ij},C,\delta,r)$ describing the haplotype frequencies among ovules and pollen, and $S$ is the proportion of all ovules produced by the population that are self-fertilized. The genotypic frequencies in the next generation among zygotes formed by self-fertilization are 
\begin{align*} \label{eq:Gprimes}
    G'_{11} &= S (o^S_{11} + o^S_{13}/4) \\
    G'_{13} &= S (o^S_{13}/2) \\
    G'_{33} &= S (o^S_{33} + o^S_{13}/4), \numberthis
\end{align*}

\noindent where $o^S_{ij}$ are functions $f(F_{ij},G_{ij},w^f_{ij},C,\delta)$ describing the proportional contribution of each genotype to self-fertilized ovules. All $G'_{ij} = 0$ where $ij \neq [11,13,33]$ (see Appendix A in the Online Supporting Information for a full derivation of all recursions). 


\subsection{Androdioecy}

\subsection{Analyses}
\begin{itemize}
	\item Gynodioecy \& Androdioecy (dominant)
	\begin{itemize}
		\item invasion analyses
		\item invasion into populations at 1-locus equilibrium.
	\end{itemize}
	\item Gynodioecy \& Androdioecy (recessive)
	\begin{itemize}
		\item Deterministic simulations
	\end{itemize}
	
\end{itemize}



\section{Results and Discussion}

The results and discussion should not be repetitive. The results section should give a factual presentation of the data and all tables and figures should be referenced; the discussion should not summarize the results but provide an interpretation of the results, and should clearly delineate between the findings of the particular study and the possible impact of those findings in a larger context. Authors are encouraged to cite recent work relevant to their interpretations. Present and discuss results only once, not in both the Results and Discussion sections. It is sometimes acceptable to combine results and discussion. The text should be as succinct as possible. Heed Strunk and White's dictum: "Omit needless words!"

\section{Additional guidelines}

\subsection{Numbers} In the text, write out numbers nine or less except as part of a date, a fraction or decimal, a percentage, or a unit of measurement. Use Arabic numbers for those larger than nine, except as the first word of a sentence; however, try to avoid starting a sentence with such a number.

\subsection{Units} Use abbreviations of the customary units of measurement only when they are preceded by a number: "3 min" but "several minutes". Write "percent" as one word, except when used with a number: "several percent" but "75\%." To indicate temperature in centigrade, use ° (for example, 37°); include a letter after the degree symbol only when some other scale is intended (for example, 45°K).

\subsection{Nomenclature and Italicization} Italicize names of organisms even when  when the species is not indicated.  Italicize the first three letters of the names of restriction enzyme cleavage sites, as in HindIII. Write the names of strains in roman except when incorporating specific genotypic designations. Italicize genotype names and symbols, including all components of alleles, but not when the name of a gene is the same as the name of an enzyme. Do not use "+" to indicate wild type. Carefully distinguish between genotype (italicized) and phenotype (not italicized) in both the writing and the symbolism.

\subsection{Cross References}
Use the \verb|\nameref| command with the \verb|\label| command to insert cross-references to section headings. For example, a \verb|\label| has been defined in the section \nameref{sec:materials:methods}.

\section{In-text Citations}

Add citations using the \verb|\citep{}| command, for example \citep{neher2013genealogies} or for multiple citations, \citep{neher2013genealogies, rodelsperger2014characterization}

\section{Examples of Article Components}
\label{sec:examples}

The sections below show examples of different header levels, which you can use in the primary sections of the manuscript (Results, Discussion, etc.) to organize your content.

\section{First level section header}

Use this level to group two or more closely related headings in a long article.

\subsection{Second level section header}

Second level section text.

\subsubsection{Third level section header:}

Third level section text. These headings may be numbered, but only when the numbers must be cited in the text. 

\section{Figures and Tables}

Figures and Tables should be labelled and referenced in the standard way using the \verb|\label{}| and \verb|\ref{}| commands.

\subsection{Sample Figure}

Figure \ref{fig:spectrum} shows an example figure.

\begin{figure}[htbp]
\centering
\includegraphics[width=\linewidth]{example-figure}
\caption{Example figure from \url{10.1534/genetics.114.173807}. Please include your figures in the manuscript for the review process. You can upload figures to Overleaf via the Project menu. Upon acceptance, we'll ask for your figure files to be uploaded in any of the following formats: TIFF (.tiff), JPEG (.jpg), Microsoft PowerPoint (.ppt), EPS (.eps), or Adobe Illustrator (.ai).  Images should be a minimum of 300 dpi in resolution and 500 dpi minimum if line art images.  RGB, CMYK, and Grayscale are all acceptable. Halftones should be high contrast with sharp detail, because some loss of detail and contrast is inevitable in the production process. Figures should be 10-20 cm in width and 1-25 cm in height. Graph axes must be exactly perpendicular and all lines of equal density.
Label multiple figure parts with A, B, etc. in bolded type, and use Arrows and numbers to draw attention to areas you want to highlight. Legends should start with a brief title and should be a self-contained description of the content of the figure that provides enough detail to fully understand the data presented. All conventional symbols used to indicate figure data points are available for typesetting; unconventional symbols should not be used. Italicize all mathematical variables (both in the figure legend and figure) , genotypes, and additional symbols that are normally italicized.  
}%
\label{fig:spectrum}
\end{figure}

\subsection{Sample Video}

Figure \ref{video:spectrum} shows how to include a video in your manuscript.

\begin{figure}[htbp]
\centering
\includegraphics[width=\linewidth]{example-figure}
\caption{Example movie (the figure file above is used as a placeholder for this example). \textit{GENETICS} supports video and movie files that can be linked from any portion of the article - including the abstract. Acceptable formats include .asf, avi, .wav, and all types of Windows Media files.   
}%
\label{video:spectrum}
\end{figure}


\subsection{Sample Table}

Table \ref{tab:fitness} shows an example table. Avoid shading, color type, line drawings, graphics, or other illustrations within tables. Use tables for data only; present drawings, graphics, and illustrations as separate figures. Histograms should not be used to present data that can be captured easily in text or small tables, as they take up much more space.  

Tables numbers are given in Arabic numerals. Tables should not be numbered 1A, 1B, etc., but if necessary, interior parts of the table can be labeled A, B, etc. for easy reference in the text.  



\section{Sample Equation}

Let $X_1, X_2, \ldots, X_n$ be a sequence of independent and identically distributed random variables with $\text{E}[X_i] = \mu$ and $\text{Var}[X_i] = \sigma^2 < \infty$, and let
\begin{equation}
S_n = \frac{X_1 + X_2 + \cdots + X_n}{n}
      = \frac{1}{n}\sum_{i}^{n} X_i
\label{eq:refname1}
\end{equation}
denote their mean. Then as $n$ approaches infinity, the random variables $\sqrt{n}(S_n - \mu)$ converge in distribution to a normal $\mathcal{N}(0, \sigma^2)$.

\bibliography{example-bibliography}

\end{document}